<!DOCTYPE html>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async
src="https://www.googletagmanager.com/gtag/js?id=UA-109009447-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

        gtag('config', 'UA-109009447-1');
</script>

<html lang="en">
<head>
<meta charset="utf-8">
<title>TopPIC Suite</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="description" content="">
<meta name="author" content="xiaowen" >

<script src="https://cdn.jsdelivr.net/npm/vega@5.20.2"></script>
<script src="https://cdn.jsdelivr.net/npm/vega-lite@5.1.0"></script>
<script src="https://cdn.jsdelivr.net/npm/vega-embed@6.17.0"></script>

<link href="assets/css/bootstrap.css" rel="stylesheet">
<link href="style.css" rel="stylesheet">
<link href="assets/css/bootstrap-responsive.css" rel="stylesheet">
<link rel="shortcut icon" href="toppic.ico">
</head>

<body style="overflow-y:scroll">
<div class="container">
  <div class="masthead">
    <h2 class="muted"><font color=#0066DD>TopPIC Suite</font></h2>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container">
          <ul class="nav">
            <li><a href="index.html">Summary</a></li>
            <li><a href="register.html">Downloads</a></li>
            <li><a href="manual.html">Manual</a></li>
            <li><a href="tutorial.html">Tutorial</a></li> 
            <li><a href="publications.html">Publications</a></li>
            <li><a href="contact.html">Contact</a></li>
          </ul>
        </div>
      </div>
    </div>
  </div>

  <div class="row-fluid">
    <div class="span-12" style="font-size:16px">
      <h3><font color=#0066DD>Summary</font></h3>

      <p>
      <b>TopPIC Suite</b>  
      consists of five software tools for the analysis of top-down mass
      spectrometry data: TopFD, TopIndex, TopPIC, TopMG, and TopDiff.
      </p>

      <ul style="list-style-type:circle">
        <li>
          <b>TopFD</b> 
          (Top-down mass spectral Feature Detection) is a software tool
          for top-down spectral deconvolution and a successor to MS-Deconv.
          It groups top-down spectral
          peaks into isotopic envelopes and converts isotopic envelopes to
          monoisotopic neutral masses. In addition, it extracts proteoform features
          from LC-MS or CE-MS data. 
        </li>

        <li>
          <b>TopIndex</b> (Top-down protein sequence database Indexing) 
          generates index files for protein sequence databases. 
          The index files are used in TopPIC and TopMG to speed up proteoform
          identification by database search. 
        </li>


        <li> 
          <b>TopPIC</b> 
          (Top-down mass spectrometry-based Proteoform Identification and
          Characterization) identifies and characterizes proteoforms at the
          proteome level by searching top-down tandem mass spectra
          against a protein sequence database. TopPIC is a successor to MS-Align+. 
          It efficiently identifies
          proteoforms with unexpected alterations, such as mutations and post-translational
          modifications (PTMs), accurately estimates the statistical significance of
          identifications, and characterizes reported proteoforms with 
          unknown mass shifts.  It uses several techniques, such as indexes,
          spectral alignment, generating function methods, and the
          modification identification score (MIScore), to increase the speed,
          sensitivity, and accuracy. 
        </li>

        <li>
          <b>TopMG</b> 
          (Top-down mass spectrometry-based proteoform identification using
          Mass Graphs) is a software tool for identifying highly modified proteoforms 
          by searching top-down tandem mass spectra against a protein sequence
          database. It is capable of identifying 
          proteoforms with multiple variable PTMs and unexpected alterations, such as
          histone proteoforms and phosphorylated ones. It uses mass
          graphs, which efficiently represent candidate proteoforms with
          multiple variable PTMs, to increase the speed and sensitivity in
          proteoform identification. In addition, approximate spectrum-based filtering 
          methods are employed for protein sequence filtering, and a Markov chain Monte Carlo 
          method (TopMCMC) is used for estimating the statistical significance of identifications. 
        </li>
        <li>
          <b>TopDiff</b> (Top-down mass spectrometry-based identification of
          Differentially expressed proteoforms) compares the abundances of
          proteoforms and finds differentially expressed proteoforms by 
          using identifications of top-down mass spectrometry data of several protein samples. 
        </li>
      </ul>

      <h3>Release notes</h3>
      <div id="bottom-div">
        <div id="release-notes">
          <b>Version 1.7 (December 15, 2023)</b>
          <ul style="list-style-type:circle">
            <li>[TopFD] Modified and set EnvCNN as default scoring function. </li>
            <li>[TopPIC] .</li>
          </ul>
          <b>Version 1.6 (December 10, 2022)</b>
          <ul style="list-style-type:circle">
            <li>[TopPIC] Added support for variable PTMs. </li>
            <li>[TopPIC] Rewrote protein sequence filtering functions.</li>
          </ul>
          <b>Version 1.5 (December 3, 2021)</b>
          <ul style="list-style-type:circle">
            <li>[TopFD] Added support for FAIMS data. </li>
            <li>[TopPIC] Rewrote the localization function.</li>
          </ul>
          <b>Version 1.4 (December 30, 2020)</b>
          <ul style="list-style-type:circle">
            <li>[TopFD] Added multiple thread support. </li>
            <li>[TopIndex] Added TopIndex.</li>
          </ul>
          <b>Version 1.3 (December 31, 2019)</b>
          <ul style="list-style-type:circle">
            <li>[TopPIC] Added spectral visualization using Html files.</li>
            <li>[TopMG] Added spectral visualization using Html files.</li>
            <li>[TopDiff] Added TopDiff.</li>
          </ul>
          <b>Version 1.2 (November 15, 2018)</b>
          <ul style="list-style-type:circle">
            <li>[TopFD] Added support for processing multiple files</li>
            <li>[TopFD] Optimized the code to speed up computation</li>
            <li>[TopPIC] Added support for data containing multiple fractions</li>
            <li>[TopMG] Added TopMG</li>
          </ul>

          <b>Version 1.1 (November 15, 2017)</b>
          <ul style="list-style-type:circle">
            <li>[TopFD] Added TopFD</li>
            <li>[TopFD] Added a GUI wrapper topfd_gui</li>
            <li>[TopPIC] Added a filtering method based on proteoform-level FDR</li>
            <li>[TopPIC] Added a GUI wrapper toppic_gui</li>
          </ul>

          <b>Version 1.0 (May 26, 2016)</b>
          <ul style="list-style-type:circle">
            <li>[TopPIC] Added the support for spectral pairs or triplets generated by the alternating fragmentation mode</li>
            <li>[TopPIC] Optimized the algorithms for protein filtering and spectral alignment</li>
            <li>[TopPIC] Added the MIScore method for mass shift characterization</li>
          </ul>

          <b>Version 0.9 (December 8, 2014)</b>
          <ul style="list-style-type:circle">
            <li>[TopPIC] First release</li>
          </ul>
        </div>
        <div id="download-graph-div"></div>
      </div>
    </div>

  </div>

</div>
<script src="assets/js/jquery-1.7.1.min.js"></script>
<!--
<script src="parseRegisterInfo.js"></script>
-->

</body>
</html>
