<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>TopPIC Suite</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="description" content="">
<meta name="author" content="xiaowen" >

<link href="assets/css/bootstrap.css" rel="stylesheet">
<link href="style.css" rel="stylesheet">
<link href="assets/css/bootstrap-responsive.css" rel="stylesheet">
<link rel="shortcut icon" href="toppic.ico">
</head>

<body style="overflow-y:scroll">

<div class="container">
  <div class="masthead">
    <h2 class="muted"><font color=#0066DD>TopPIC Suite</font></h2>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container">
          <ul class="nav">
            <li><a href="index.html">Summary</a></li>
            <li><a href="register.html">Downloads</a></li>
            <li><a href="manual.html">Manual</a></li>
            <li><a href="tutorial.html">Tutorial</a></li>
            <li><a href="publications.html">Publications</a></li>
            <li><a href="contact.html">Contact</a></li>
          </ul>
        </div>
      </div>
    </div><!-- /.navbar -->
  </div><!-- /.masthead -->

  <div class="row-fluid" style="font-size:16px">
    <div class="span-12">
      <h3>1 Overview</h3>
      To run TopPIC suite, a computer with at least 4 GB memory and a 64-bit
      Linux or Windows operating system is required. TopFD, TopPIC and TopMG
      provide a command line interface for both Linux and Windows users as well
      as a graphical user interface (GUI) for Windows users. Please see our <a href="tutorial.html">tutorial</a>
      for more details on the GUI.

      <h3>2 TopFD</h3>
      <p>
      TopFD (Top-down mass spectral Feature Detection) is a software tool
      for top-down spectral deconvolution, which groups top-down mass spectral
      peaks into isotopomer envelopes and converts isotopomer envelopes to
      monoisotopic neutral masses. In addition, it extracts proteoform features
      from MS1 spectra. 
      </p>


      <h4>2.1 Input</h4>
      <p>The input of TopFD is mass spectrometry data
      files in the mzXML or mzML format. Raw mass spectral data generated from various mass spectrometers
      can be converted to mzML or mzXML files using <a target="_blank" href="http://proteowizard.sourceforge.net/tools.shtml">msconvert</a>.


      <h4>2.2 Output</h4>
      <p>TopFD outputs an LC/MS feature text file with a file extension "<b>feature</b>" and two deconvoluted mass spectral
      data files (one is for MS1 spectra and the other for MS/MS
      spectra) in the msalign format with a file extension "<b>msalign</b>", which is similar to the MGF file format. 
     
      For example, when the input file name is <b>spectra.mzML</b>, the output includes: 
      <ul>
        <li><b>spectra.feature</b>: a feature file containing MS/MS scan ids and their
        corresponding LC/MS features.</li>
        <li><b>spectra_ms1.msalign</b>: a list of deconvoluted MS1 spectra.</li>
        <li><b>spectra_ms2.msalign</b>: a list of deconvoluted MS/MS spectra.</li>
      </ul>

      </p>

      <h4>2.3 Command line usage</h4>
      <p>To run TopFD, open a terminal window and run the following command.</p>
      <p><code>topfd [options] spectrum-file-names</code></p>

      <h4>Options</h4>
      <p><code>-h [ --help ]</code></p>
      <p>Print the help message.</p>

      <p><code>-c [ --max-charge ] &lt;a positive integer&gt;</code></p>
      <p>Set the maximum charge state of precursor and fragment ions. The default value is 30.</p>

      <p><code>-m [ --max-mass ] &lt;a positive number&gt;</code></p>
      <p>Set the maximum monoisotopic mass of precursor and fragment ions. The
      default value is 100,000 Dalton.</p>

      <p><code>-e [ --mz-error ] &lt;a positive number&gt;</code></p>
      <p>Set the error tolerance of m/z values of spectral peaks. The default
      value is 0.02 m/z.</p>

      <p><code>-s [ --ms-two-sn-ratio ] &lt;a positive number&gt;</code></p>
      <p>Set the signal/noise ratio for MS/MS spectra. The default value is 1.</p>

      <p><code>-r [ --ms-one-sn-ratio ] &lt;a positive number&gt;</code></p>
      <p>Set the signal/noise ratio for MS1 spectra. The default value is 3.</p>

      <p><code>-w [ --precursor-window ] &lt;a positive number&gt;</code></p>
      <p>Set the precursor isolation window size. The default value is 3.0 m/z.</p>

      <p><code>-n [ --missing-level-one ]</code></p>
      <p>Specify that the input file does not contain MS1 spectra.</p>

      <h4>Examples</h4>
      <p>
      Deconvolute a centroid data file spectra.mzML and output three files:
      spectra_ms1.msalign, spectra_ms2.msalign, and spectra.feature.
		</p>
      <p><code> topfd spectra.mzML </code></p>
      
      <p>
      Deconvolute two centroid data files spectra1.mzML and spectra2.mzML and 
      output three files for each input data file.
		</p>
      <p><code> topfd spectra1.mzML spectra2.mzML </code></p>
      
      <p>
      Deconvolute all centroid data files in the current folder. 
		</p>
		
      <p><code> topfd *.mzML </code></p>

      <p>
      Deconvolute a centroid data file spectra.mzML that does not contain MS1
      spectra. 
		</p>
      <p><code> topfd -n spectra.mzML </code></p>
      
      <p>
      Deconvolute a centroid data file spectra.mzML with a precursor isolation
      window size 2 m/z.  
		</p>
      <p><code> topfd -w 2 spectra.mzML </code></p>
      
      <p>
      Deconvolute a centroid data file spectra.mzML with a signal/noise ratio 2 for MS1 spectra. 
		</p>
      <p><code> topfd -r 2 spectra.mzML </code></p>

      <p>
      Deconvolute a centroid data file spectra.mzML with the following settings: 
      set the charge state to 50, the maximum mass to 30,000 Dalton, and the 
      signal/noise ratio to 2 for MS/MS spectra.   
		</p>
      <p><code> topfd -c 50 -m 30000 -s 2 spectra.mzML </code></p>
      <p>
<!--
      <h4>2.4 Graphical user interface</h4>
      <p>
      To run the GUI of TopFD, double-click the downloaded executive file
      <code>topfd_gui.exe</code>. To deconvolute a spectrum data file,
      use the button "File" to select a data file, change parameters
      if needed, and click the button "Start" to start deconvolution. After
      TopFD completes the deconvolution of the file, click the button "Output
      folder" to find output files.
-->

      <h3>3 TopPIC</h3>

      <h4>3.1 Input</h4>
      <ul>
        <li>A protein database file in the FASTA format</li>
        <li>A mass spectrum data file in the msalign format</li>
        <li>A text file containing LC/MS feature information (optional)</li>
        <li>A text file specifying fixed PTMs (optional)</li>
        <li>A text file specifying PTMs for the characterization of unexpected mass shifts
        (optional)</li>
      </ul>

      <h4>3.2 Output</h4>
      <p>TopPIC outputs two comma separated value files, an xml file, and a collection of html files for identified
      proteoforms. For example, when the input data file is
      spectra_ms2.msalign, the output includes: 
      <ul>
        <li><b>spectra_ms2_toppic_prsm.csv</b>: a comma separated value file containing 
        identified proteoform spectrum-matches (PrSMs) with an E-value or spectrum level 
        FDR cutoff.</li>
        <li><b>spectra_ms2_toppic_proteoform.csv</b>: a comma separated value file containing 
        identified proteoforms with an E-value or proteoform level FDR cutoff.</li>
        <li><b>spectra_ms2_toppic_proteoform.xml</b>: an xml file containing 
        identified proteoforms with an E-value or proteoform level FDR cutoff.</li>
        <li><b>spectra_ms2_toppic_prsm_cutoff_html</b>: a folder containing html files for the
        annotation of identified PrSMs using the 
        E-value or spectrum level FDR cutoff.</li>
        <li><b>spectra_ms2_toppic_proteoform_cutoff_html</b>: a folder containing html files for the
        annotation of identified PrSMs using the E-value or 
        proteoform level FDR cutoff.</li>
      </ul>

      To browse identified
      proteins, proteoforms, and PrSMs,  
      use a web browser to open the file spectrum_ms2_toppic_prsm_cutoff_html/proteins.html 
      or spectrum_ms_toppic_proteoform_cutoff_html/proteins.html.
      Google Chrome or Firefox is recommended (IE is not recommended).
      </p>
      <p>When the input contains two or more data files,
      TopPIC outputs two tab delimited 
      text files, an xml file, and a collection of html files for each input file. 
      When a file name is specified for combined identifications, it combines 
      spectra and proteoforms identified from all the input files, removes redundant proteoform 
      identifications,and reports 
      two comma separated value files, an xml file, and a collection of html files for the combined results. 
      For example, when the input is spectra1_ms2.msalign and spectra2_ms2.msalign and the combined
      output file name is "combined", the output files are:
      
      <ul>
        <li><b>combined_toppic_prsm.csv</b>: a comma separated value file containing 
        PrSMs identified from all the input files with an E-value or spectrum level 
        FDR cutoff.</li>
        <li><b>combined_toppic_proteoform.csv</b>: a comma separated value file containing 
        proteoforms identified from all the input files with an E-value or proteoform level FDR cutoff.</li>
        <li><b>combined_toppic_proteoform.xml</b>: an xml file containing 
        proteoforms identified from all the input files with an E-value or proteoform level FDR cutoff.</li>
        <li><b>combined_toppic_prsm_cutoff_html</b>: a folder containing html files for the
        annotation of PrSMs identified from all the input files using the E-value or  
        spectrum level FDR cutoff.</li>
        <li><b>combined_toppic_proteoform_cutoff_html</b>: a folder containing html files for the
        annotation of PrSMs identified from all the input files using the 
        E-value or proteoform level FDR cutoff.</li>
      </ul>
       
      </p>
      <h4>3.3 Command line usage</h4>
      <p>To run TopPIC, open a terminal window and run the following command.</p>
      <p><code>toppic [options] database-file-name spectrum-file-names</code></p>

      <h4>Options</h4>
      <p><code>-h [ --help ]</code></p>
      <p>Print the help message.</p>

      <p><code>-a [ --activation ] &lt;CID|HCD|ETD|UVPD|FILE&gt;</code></p>
      <p>Set the fragmentation method(s) of MS/MS spectra. When "FILE" is
      selected, 
      the fragmentation methods of spectra are given in the input spectrum data
      file. Default value: FILE.</p>

      <p><code>-f [ --fixed-mod ] &lt;C57|C58|a fixed modification file&gt;</code></p>
      <p>Set fixed modifications. Three available options: C57, C58, or the name of
      a text file containing the information of fixed modifications (see an <a
        target="_blank" href="example_files/fixed_mods.txt">example file</a>).
      When C57 is selected, carbamidomethylation on cysteine is the only fixed
      modification.  When C58 is selected, carboxymethylation on cysteine is the
      only fixed modification.</p>

      <p><code>-n [ --n-terminal-form ] &lt;a list of allowed N-terminal forms&gt;</code></p>
      <p>Set N-terminal forms of proteins. Four N-terminal forms can be selected:
      NONE, NME, NME_ACETYLATION, and M_ACETYLATION.  NONE stands for no
      modifications, NME for N-terminal methionine excision, NME_ACETYLATION for
      N-terminal acetylation after the initiator methionine is removed, and
      M_ACETYLATION for N-terminal methionine acetylation.  When multiple forms
      are allowed, they are separated by commas. Default value:
      NONE,M_ACETYLATION,NME,NME_ACETYLATION.</p>

      <p><code>-d [ --decoy ]</code></p>
      <p>Use a shuffled decoy protein database to estimate spectrum and
      proteoform level FDRs.
      When -d is chosen, a shuffled decoy database is automatically
      generated and appended to the target database before database search, and
      FDR rates are estimated using the target-decoy approach.</p>

      <p><code>-e [ --error-tolerance ] &lt;a positive integer&gt;</code></p>
      <p>Set the error tolerance for precursor and fragment masses in part-per-million
      (ppm). Default value: 15. When the default lookup table approach is used
      for E-value estimation, valid error tolerance values are 5, 10, and 15 ppm. To use other
      values, the generating function approach (<code>-g</code>) should be chosen.</p>

      <p><code>-m [ --max-shift ]  &lt;a number&gt;</code></p>
      <p>Set the maximum value for unexpected mass shifts (in Dalton).
      Default value: 500.</p>

      <p><code>-M [ --min-shift ]  &lt;a number&gt;</code></p>
      <p>Se the minimum value for unexpected mass shifts (in Dalton). 
      Default value: -500.</p>

      <p><code>-p [ --num-shift ] &lt;0|1|2&gt;</code></p>
      <p>Set the maximum number of unexpected mass shifts in a PrSM. Default value: 1.</p>

      <p><code>-t [ --spectrum-cutoff-type ] &lt;EVALUE|FDR&gt;</code></p>
      <p>Set the spectrum level cutoff type for filtering PrSMs. Default value: EVALUE.</p>

      <p><code>-v [ --spectrum-cutoff-value ] &lt;a positive number&gt;</code></p>
      <p>Set the spectrum level cutoff value for filtering PrSMs. Default value: 0.01.</p>

      <p><code>-T [ --proteoform-cutoff-type ] &lt;EVALUE|FDR&gt;</code></p>
      <p>Set the proteoform level cutoff type for filtering proteoforms and
      PrSMs. Default value: EVALUE.</p>

      <p><code>-V [ --proteoform-cutoff-value ] &lt;a positive number&gt;</code></p>
      <p>Set the proteoform level cutoff value for filtering proteoforms and
      PrSMs. Default value: 0.01.</p>

      <p><code>-b [ --lookup-table ]</code></p>
      <p>Use a lookup table method for computing p-values and E-values.
      It is faster than the default generating function approach, but it
      may reduce the number of identifications.</p>

      <p><code>-r [ --num-combined-spectra ] &lt;a positive integer&gt;</code></p>
      <p>Set the number of combined spectra. The parameter is set to 2 (or 3) for combining spectral pairs (or triplets) 
      generated by the alternating fragmentation mode. Default value: 1.</p>

      <p><code>-i [ --mod-file-name ] &lt;a common modification file&gt; </code></p>
      <p>Specify a text file containing a list of common PTMs for proteoform characterization. The PTMs are used to identify and
      localize PTMs in reported PrSMs with unknown mass
      shifts. See an <a target="_blank" href="example_files/common_mods.txt" >example file</a>.</p>

      <p><code>-s [ --miscore-threshold ] &lt;a number between 0 and 1&gt;</code></p>
      <p>Set the score threshold (MIScore) for filtering results of PTM characterization. Default value: 0.45.</p>

      <p><code>-u [ --thread-number ] &lt;a positive number&gt;</code></p>
      <p>Set the number of threads used in the computation. Default value: 1. The maximum number of threads is determined by the CPU 
      and memory of the computer used for computation. About 4 GB memory is required for each thread. If the computer has 16 GB memory
      and a CPU with 8 cores, the maximum number of threads is 4 because 4 threads need about 16 GB memory.</p>

      <p><code>-x [ --no-topfd-feature ]</code></p>
      <p>Specify that there are no TopFD feature files for proteoform identification.</p>

      <p><code>-l [ --skip-list ] &lt;a text file with its path&gt;</code></p>  
      <p>Specify a text file containing a list of spectrum scans that have been
      identified in previous analyses and are not needed to be searched against
      the protein database. 
      See <a target="_blank" href="example_files/scan_list.txt" >example file</a>.</p>
      </p> 

      <p><code>-c [ --combined-file-name ] &lt;a filename&gt;</code></p>
      <p>Specify an output file name for combined identifications when the input consists of multiple spectrum files. 
      </p>

      <p><code>-k [ --keep ]</code></p>  
      <p>Keep intermediate files generated by TopPIC. 
      </p> 

      <h4>Examples</h4>
      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta with a feature file spectra.feature. 
      The user does not need to specify the feature file name. TopPIC will automatically 
      obtain the name spectra.feature from the spectrum file name spectra_ms2.msalign.</p>
      <p><code> toppic proteins.fasta spectra_ms2.msalign </code></p>
      
      <p> Search two deconvoluted MS/MS spectrum files spectra1_ms2.msalign and 
      spectra2_ms2.msalign against a protein database file proteins.fasta with feature files. 
      In addition, all identifications are combined and reported using a file name "combined." 
      </p>
      <p><code> toppic -c combined proteins.fasta spectra1_ms2.msalign spectra2_ms2.msalign </code></p>

      <p> Search all deconvoluted MS/MS spectrum files in the current folder
      against a protein database file proteins.fasta with feature files. 
      </p>
      <p><code> toppic proteins.fasta *_ms2.msalign </code></p>
      
      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta without feature files.</p>
      <p><code> toppic -x proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta with a feature file spectra.feature
      and a fixed modification: carbamidomethylation on cysteine. </p>
      <p><code> toppic -f C57 proteins.fasta spectra_ms2.msalign </code></p>
      
      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file
      spectra.feature.  In an identified proteoform, at most 2 mass shifts are
      allowed and the maximum allowed mass shift value is 10,000 Dalton. </p>
      <p><code> toppic -p 2 -m 10000 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file
      spectra.feature.  The error tolerance for precursor and fragment masses is
      5 ppm. </p> 
      <p><code> toppic -e 5 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file
      spectra.feature. Use the target decoy approach to compute spectrum level
      and proteoform level FDRs, filter identified proteoform
      spectrum-matches by a 5% spectrum level FDR, and filter identified
      proteoforms by a 5% proteoform level FDR. 
      <p><code> toppic -d -t FDR -v 0.05 -T FDR -V 0.05 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign with
      alternating CID, HCD, and ETD spectra against a protein database file
      proteins.fasta with a feature file spectra.feature. Combine alternating
      CID, HCD, and ETD spectra to increase proteoform coverage. 
      <p><code> toppic -r 3 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file
      spectra.feature. After proteoforms with unexpected mass shifts are
      identified, match these mass shifts to four common PTMs: acetylation,
      phosphorylation, oxidation and methylation, and use an MIScore cutoff 0.3
      to filter reported PTM sites. The modification file common_mods.txt can be 
      found <a target="_blank" href="example_files/common_mods.txt">here</a>.</p>
      <p><code> toppic -i common_mods.txt -s 0.3 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file
      spectra.feature. Use 6 CPU threads to speed up the computation. </p> 
      <p><code> toppic -u 6 proteins.fasta spectra_ms2.msalign </code></p>



      <h3>4 TopMG</h3>

      <h4>4.1 Input</h4>
      <ul>
        <li>A protein database file in the FASTA format</li>
        <li>A mass spectrum data file in the msalign format</li>
        <li>A text file specifying variable PTMs</li>
        <li>A text file specifying fixed PTMs (optional)</li>
        <li>A text file containing LC/MS feature information (optional)</li>
      </ul>

        
		<h4>4.2 Output</h4>
      <p>TopMG outputs two comma separated value files, an xml file, and a collection of html files for identified
      proteoforms. For example, when the input mass spectrum data file is
      spectra_ms2.msalign, the output includes: 
      <ul>
        <li><b>spectra_ms2_topmg_prsm.csv</b>: a comma separated value file containing 
        identified PrSMs with an E-value or spectrum level 
        FDR cutoff.</li>
        <li><b>spectra_ms2_topmg_proteoform.csv</b>: a comma separated value file containing 
        identified proteoforms with an E-value or proteoform level FDR cutoff.</li>
        <li><b>spectra_ms2_topmg_proteoform.xml</b>: an xml file containing 
        identified proteoforms with an E-value or proteoform level FDR cutoff.</li>
        <li><b>spectra_ms2_topmg_prsm_cutoff_html</b>: a folder containing html files for the
        annotation of identified PrSMs using the E-value or
        spectrum level  FDR cutoff.</li>
        <li><b>spectra_ms2_topmg_proteoform_cutoff_html</b>: a folder containing html files for the
        annotation of identified PrSMs using the E-value or  
        proteoform level cutoff.</li>
      </ul>

      To browse identified
      proteins, proteoforms, and PrSMs,  
      use a web browser to open the file spectrum_ms2_topmg_prsm_cutoff_html/proteins.html 
      or spectrum_ms_topmg_proteoform_cutoff_html/proteins.html.
      Google Chrome or Firefox is recommended (IE is not recommended).
      </p>
      <p>When the input contains two or more spectrum files, TopMG outputs two tab delimited 
      text files, an xml file, and a collection of html files for each input file. 
      When a file name is specified for combined identifications, it combines 
      spectra and proteoforms identified from all the input files, removes redundant proteoform 
      identifications,and reports 
      two comma separated value files, an xml file, and a collection of html files for the combined results. 
      
      For example, when the input is spectra1_ms2.msalign and spectra2_ms2.msalign and the combined
      output file name is "combined", the output files are:      

      <ul>
        <li><b>combined_topmg_prsm.csv</b>: a comma separated value file containing 
        PrSMs identified from all the input files with an E-value or spectrum level
        FDR cutoff.</li>
        <li><b>combined_topmg_proteoform.csv</b>: a comma separated value file containing 
        proteoforms identified from all the input files with an E-value or proteoform level FDR cutoff.</li>
        <li><b>combined_topmg_proteoform.xml</b>: an xml file containing 
        proteoforms identified from all the input files with an E-value or proteoform level FDR cutoff.</li>
        <li><b>combined_topmg_prsm_cutoff_html</b>: a folder containing html files for the
        annotation of PrSMs identified from all the input files using the E-value or 
        spectrum level FDR cutoff.</li>
        <li><b>combined_topmg_proteoform_cutoff_html</b>: a folder containing html files for the
        annotation of PrSMs identified from all the input files using the 
        E-value or proteoform level cutoff.</li>
      </ul>      
      
      <h4>4.3 Command line usage</h4>
      <p>To run TopMG, open a terminal window and run the following command.</p>
      <p><code>topmg [options] database-file-name spectrum-file-names</code></p>

      <h4>Options</h4>
      <p><code>-h [ --help ]</code></p>
      <p>Print the help message.</p>

      <p><code>-a [ --activation ] &lt;CID|HCD|ETD|UVPD|FILE&gt;</code></p>
      <p>Fragmentation method of tandem mass spectra. When FILE is used, 
        fragmentation methods of spectra are given in the input spectral data file. 
        Default value: FILE.</p>

      <p><code>-f [ --fixed-mod ] &lt;C57|C58|a fixed modification file&gt;</code></p>
      <p>Set fixed modifications. Three available options: C57, C58, or the name of
      a text file specifying fixed modifications (see an <a
        target="_blank" href="example_files/fixed_mods.txt">example file</a>).
      When C57 is selected, carbamidomethylation on cysteine is the only fixed
      modification.  When C58 is selected, carboxymethylation on cysteine is the
      only fixed modification.</p>

      <p><code>-n [ --n-terminal-form ] &lt;a list of allowed N-terminal forms&gt;</code></p>
      <p>Set N-terminal forms of proteins. Four N-terminal forms can be selected:
      NONE, NME, NME_ACETYLATION, and M_ACETYLATION.  NONE stands for no
      modifications, NME for N-terminal methionine excision, NME_ACETYLATION for
      N-terminal acetylation after the initiator methionine is removed, and
      M_ACETYLATION for N-terminal methionine acetylation.  When multiple forms
      are allowed, they are separated by commas. Default value:
      NONE,M_ACETYLATION,NME,NME_ACETYLATION.</p>

      <p><code>-d [ --decoy ]</code></p>
      <p>Use a shuffled decoy protein database to estimate spectrum and
      proteoform level FDRs.
      When -d is chosen, a shuffled decoy database is automatically
      generated and appended to the target database before database search, and
      FDR rates are estimated using the target-decoy approach.</p>

      <p><code>-e [ --error-tolerance ] &lt;a positive integer&gt;</code></p>
      <p>Set the error tolerance for precursor and fragment masses in ppm. Default value: 15.</p>

      <p><code>-m [ --max-shift ]  &lt;a number&gt;</code></p>
      <p>Set the maximum absolute value for unexpected mass shifts (in Dalton). Default value: 500.</p>

      <p><code>-t [ --spectrum-cutoff-type ] &lt;EVALUE|FDR&gt;</code></p>
      <p>Set the spectrum level cutoff type for filtering PrSMs. Default value: EVALUE.</p>

      <p><code>-v [ --spectrum-cutoff-value ] &lt;a positive number&gt;</code></p>
      <p>Set the spectrum level cutoff value for filtering PrSMs. Default value: 0.01.</p>

      <p><code>-T [ --proteoform-cutoff-type ] &lt;EVALUE|FDR&gt;</code></p>
      <p>Set the proteoform level cutoff type for filtering proteoforms and
      PrSMs. Default value: EVALUE.</p>

      <p><code>-V [ --proteoform-cutoff-value ] &lt;a positive number&gt;</code></p>
      <p>Set the proteoform level cutoff value for filtering proteoforms and
      PrSMs. Default value: 0.01.</p>

      <p><code>-i [ --mod-file-name ] &lt;a modification file&gt; </code></p>
      <p>Specify a text file specifying variable PTMs. 
        See an <a target="_blank" href="example_files/variable_mods.txt" >example file</a>.</p>

      <p><code>-u [ --thread-number ] &lt;a positive number&gt;</code></p>
      <p>Set the number of threads used in the computation. Default value: 1.
			The maximum number of threads is determined by the CPU and memory 
			of the computer used for computation. About 4 GB memory is required for each thread. 
			If the computer has 16 GB memory and a CPU with 8 cores, the maximum number 
			of threads is 4 because 4 threads need about 16 GB memory.      
      </p>

      <p><code>-x [ --no-topfd-feature ]</code></p>
      <p>Specify that there are no TopFD feature files for proteoform identification.</p>

      <p><code>-l [ --skip-list ] &lt;a text file with its path&gt;</code></p>  
      <p>Specify a text file containing a list of spectrum scans that have been
      identified in previous analyses and are not needed to be searched against
      the protein database. 
      See an <a target="_blank" href="example_files/scan_list.txt" >example file</a>.</p>
      </p> 

      <p><code>-D [ --use-asf-diagonal ]</code></p>
      <p>Use the ASF-DIAGONAL method for protein sequence filtering. 
      The default filtering method is ASF-RESTRICT. When -D is set, both ASF-RESTRICT 
      and ASF-DIAGONAL will be used. The combined approach may identify more PrSMs, 
      but it is much slower than using ASF-RESTRICT only. See 
		<a target="_blank" href="https://www.ncbi.nlm.nih.gov/pubmed/29327814">this paper</a>    
      for more details.</p>

      <p><code>-P [ --var-ptm ] &lt;a positive number&gt;</code></p>
      <p>Set the maximum number of variable PTM sites in a proteoform. Default value: 5.</p>

      <p><code>-p [ --num-shift  &lt;0|1|2&gt;</code></p>
      <p>Set the maximum number of unexpected mass shifts in a proteoform. Default value: 0.</p>

      <p><code>-c [ --combined-file-name ] &lt;a filename&gt;</code></p>
      <p>Specify an output file name for combined identifications when the input consists of multiple spectrum files. 
      </p>
      
      <p><code>-k [ --keep ]</code></p>  
      <p>Keep intermediate files generated by TopMG. 
      </p>
      
      <h4>Advanced options</h4>
      <p><code>-j [ --proteo-graph-dis ] &lt;a positive number&gt;</code></p>
      <p>Set the length of the largest gap in constructing proteoform graphs. Default value: 40. 
      See 
      <a target="_blank" href="https://www.ncbi.nlm.nih.gov/pubmed/28453668">this paper</a>    
      for more details.</p>
      </p>

		<p><code>-G [ --var-ptm-in-gap ] &lt;a positive number&gt;</code></p>
      <p>Set the maximum number of variable PTM sites in a gap in a proteoform graph. Default value: 5.
      See 
      <a target="_blank" href="https://www.ncbi.nlm.nih.gov/pubmed/28453668">this paper</a>    
      for more details.</p>
      </p>

      <h4>Examples</h4>
      <p> To use the following examples, a variable modification file variable_mods.txt is needed in the current folder. (See an  
      <a target="_blank" href="example_files/variable_mods.txt" >example</a>.)
      </p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta with a feature file spectra.feature. 
      The user does not need to specify the feature file name. TopMG will automatically 
      obtain the name spectra.feature from the spectrum file name spectra_ms2.msalign.</p>
      <p><code> topmg -i variable_mods.txt proteins.fasta spectra_ms2.msalign </code></p>
      
      <p> Search two deconvoluted MS/MS spectrum files spectra1_ms2.msalign and 
      spectra2_ms2.msalign against a protein database file proteins.fasta with feature files.
      In addition, all identifications are combined and reported using a file name "combined." 
      </p>
      <p><code> topmg -i variable_mods.txt -c combined proteins.fasta spectra1_ms2.msalign spectra2_ms2.msalign </code></p>

      <p> Search all deconvoluted MS/MS spectrum files in the current folder
      against a protein database file proteins.fasta with feature files. 
      </p>
      <p><code> topmg -i variable_mods.txt proteins.fasta *_ms2.msalign </code></p>
      
      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta without feature files.</p>
      <p><code> topmg -i variable_mods.txt -x proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta with a feature file spectra.feature
      and a fixed modification: carbamidomethylation on cysteine. </p>
      <p><code> topmg -i variable_mods.txt -f C57 proteins.fasta spectra_ms2.msalign </code></p>
      
      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file
      spectra.feature.  In an identified proteoform, at most 1 unexpected mass shift and 
      4 variable PTMs are
      allowed and the maximum value for unexpected mass shifts is 10,000 Dalton. </p>
      <p><code> topmg -i variable_mods.txt -P 4 -p 1 -m 10000 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file
      spectra.feature.  The error tolerance for precursor and fragment masses is
      5 ppm. </p> 
      <p><code> topmg -i variable_mods.txt -e 5 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file
      spectra.feature. Use the target decoy approach to compute spectrum level
      and proteoform level FDRs, filter identified proteoform
      spectrum-matches by a 5% spectrum level FDR, and filter identified
      proteoforms by a 5% proteoform level FDR. 
      <p><code> topmg -i variable_mods.txt -d -t FDR -v 0.05 -T FDR -V 0.05 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file
      spectra.feature. Use 6 CPU threads to speed up the computation. </p> 
      <p><code> topmg -i variable_mods.txt -u 6 proteins.fasta spectra_ms2.msalign </code></p>      

      <h3>5 TopMerge</h3>

      <h4>5.1 Input</h4>
      <ul>
        <li>A protein database file in the FASTA format</li>
        <li>Several proteoform identification files in the xml format</li>
      </ul>

        
		<h4>5.2 Output</h4>
      <p>TopMG outputs a tab delimited text file containing the abundances of the proteoforms reported in the identification files. 
      The default output file name is sample_merged.csv.
      </p> 
      
      <h4>5.3 Command line usage</h4>
      <p>To run TopMerge, open a terminal window and run the following command.</p>
      <p><code>topmerge [options] database-file-name identification-file-names</code></p>

      <h4>Options</h4>
      <p><code>-h [ --help ]</code></p>
      <p>Print the help message.</p>

      <p><code>-f [ --fixed-mod ] &lt;C57|C58|a fixed modification file&gt;</code></p>
      <p>Set fixed modifications. Three available options: C57, C58, or the name of
      a text file specifying fixed modifications (see an <a
        target="_blank" href="example_files/fixed_mods.txt">example file</a>).
      When C57 is selected, carbamidomethylation on cysteine is the only fixed
      modification.  When C58 is selected, carboxymethylation on cysteine is the
      only fixed modification. The setting of fixed modifications should be the same as that 
      in proteoform identification.</p>
      
      <p><code>-e [ --error-tolerance ] &lt;a positive integer&gt;</code></p>
      <p>Set the error tolerance for mapping identified proteoforms in multiple samples. Default value: 1.2 Dalton.</p>

      <p><code>-o [ --output ] &lt;a filename&gt;</code></p>
      <p>Specify an output file name for merged proteoform identifications of multiple samples. The default file name is sample_merged.csv. 
      </p>
      
      <h4>Examples</h4>

      <p> Merge two proteoform identification files spectra1_ms2_toppic_proteoform.xml and spectra2_ms2_toppic_proteoform.xml
      reported from searches against a protein database proteins.fasta. </p>
      <p><code> topmerge proteins.fasta spectra1_ms2_toppic_proteoform.xml spectra2_ms2_toppic_proteoform.xml </code></p>
      
      <p> Merge two proteoform identification files spectra1_ms2_toppic_proteoform.xml and spectra2_ms2_toppic_proteoform.xml
      reported from searches against a protein database proteins.fasta. The fixed modification was set to C57 in database searches.</p>
      <p><code> topmerge -f C57 proteins.fasta spectra1_ms2_toppic_proteoform.xml spectra2_ms2_toppic_proteoform.xml </code></p>
      
      <p> Merge two proteoform identification files spectra1_ms2_toppic_proteoform.xml and spectra2_ms2_toppic_proteoform.xml
      reported from searches against a protein database proteins.fasta. Set the error tolerance for mapping identified proteoforms to 0.2 Dalton.</p>
      <p><code> topmerge -e 0.2 proteins.fasta spectra1_ms2_toppic_proteoform.xml spectra2_ms2_toppic_proteoform.xml </code></p>
      
      <p> Merge two proteoform identification files spectra1_ms2_toppic_proteoform.xml and spectra2_ms2_toppic_proteoform.xml
      reported from searches against a protein database proteins.fasta. Set the output filename to proteoform_quant.tsv.</p>
      <p><code> topmerge -o proteoform_quant.tsv proteins.fasta spectra1_ms2_toppic_proteoform.xml spectra2_ms2_toppic_proteoform.xml </code></p>
      
      <p> Merge all proteoform identification files reported from searches against a protein database proteins.fasta by TopPIC in the current folder. </p>
      <p><code> topmerge proteins.fasta *_ms2_toppic_proteoform.xml </code></p>
    
    </div>
  </div>
</div>
<script src="assets/js/jquery-1.7.1.min.js"></script>

</body>
</html>
