<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>TopPIC Suite</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="description" content="">
<meta name="author" content="xiaowen" >

<link href="assets/css/bootstrap.css" rel="stylesheet">
<link href="style.css" rel="stylesheet">
<link href="assets/css/bootstrap-responsive.css" rel="stylesheet">
<link rel="shortcut icon" href="toppic.ico">
</head>

<body style="overflow-y:scroll">

<div class="container">
  <div class="masthead">
    <h2 class="muted"><font color=#0066DD>TopPIC Suite</font></h2>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container">
          <ul class="nav">
            <li><a href="index.html">Summary</a></li>
            <li><a href="register.html">Downloads</a></li>
            <li><a href="manual.html">Manual</a></li>
            <li><a href="tutorial.html">Tutorial</a></li>
            <li><a href="publications.html">Publications</a></li>
            <li><a href="contact.html">Contact</a></li>
          </ul>
        </div>
      </div>
    </div><!-- /.navbar -->
  </div><!-- /.masthead -->

  <div class="row-fluid" style="font-size:16px">
    <div class="span-12">
      <h3>1 Overview</h3>
      To run TopPIC suite, a computer with at least 4 GB memory and a 64-bit
      Linux or Windows operating system is required. TopFD, TopIndex, TopPIC,
      TopMG, and TopDiff  
      provide a command line interface for both Linux and Windows users as well
      as a graphical user interface (GUI) for Windows users. Please see our <a
         href="tutorial.html">tutorials</a> for more details of the GUI.

      <h3>2 TopFD</h3>
      <p>
      TopFD (Top-down mass spectral Feature Detection) is a software tool
      for top-down spectral deconvolution, which groups top-down mass spectral
      peaks into isotopic envelopes and converts isotopic envelopes to
      monoisotopic neutral masses. In addition, it extracts proteoform features
      from MS1 spectra. 
      </p>


      <h4>2.1 Input</h4>
      <p>The input of TopFD is mass spectrometry data
      files in the mzXML or mzML format. Raw mass spectral data generated from various mass spectrometers
      can be converted to mzML or mzXML files using <a target="_blank" href="http://proteowizard.sourceforge.net/tools.shtml">msconvert</a>.


      <h4>2.2 Output</h4>
      <p>TopFD outputs two LC-MS feature text files with a file extension "<b>feature</b>", 
      one LC-MS feature file with a file extension "<b>xml</b>", 
      and a deconvoluted mass spectral data file for MS/MS spectra in the msalign 
      format with a file extension "<b>msalign</b>", which is similar to the MGF file format. 
      In addition, TopFD creates two folders containing JavaScript files of
      spectral data for spectral visualization and other files generated in spectral deconvolution.

      For example, when the input file name is <b>spectra.mzML</b>, the output includes: 
      <ul>
        <li><b>spectra_ms1.feature</b>: a feature file containing LC-MS
        features.</li>
        <li><b>spectra_ms2.feature</b>: a feature file containing MS/MS scan IDs and their
        corresponding LC-MS feature IDs.</li>
        <li><b>spectra_feature.xml</b>: a feature file containing LC-MS features in the xml format.</li>
        <li><b>spectra_ms2.msalign</b>: a list of deconvoluted MS/MS spectra.</li>
        <li><b>spectra_file</b>: a folder containing deconvoluted MS1 spectra
        and annotations of LC-MS data.</li>
        <li><b>spectra_html</b>: a folder containing JavaScript files of MS1
        and MS/MS data for spectral visualization.</li>

      </ul>

      </p>

      <h4>2.3 Command line usage</h4>
      <p>To run TopFD, open a terminal window and run the following command.</p>
      <p><code>topfd [options] spectrum-file-names</code></p>

      <h4>Options</h4>
      <p><code>-h [ --help ]</code></p>
      <p>Print the help message.</p>

      <p><code>-a [ --activation ] &lt;CID|HCD|ETD|UVPD|FILE&gt;</code></p>
      <p>Set the fragmentation method(s) of MS/MS spectra. When "FILE" is
      selected, 
      the fragmentation methods of spectra are given in the input spectrum data
      file. Default value: FILE.</p>
      
      <p><code>-c [ --max-charge ] &lt;a positive integer&gt;</code></p>
      <p>Set the maximum charge state of precursor and fragment ions. The default value is 30.</p>

      <p><code>-m [ --max-mass ] &lt;a positive number&gt;</code></p>
      <p>Set the maximum monoisotopic mass of precursor and fragment ions. The
      default value is 70,000 Dalton.</p>

      <p><code>-t [ --mz-error ] &lt;a positive number&gt;</code></p>
      <p>Set the error tolerance of m/z values of spectral peaks. The default
      value is 0.02 m/z.</p>

      <p><code>-r [ --ms-one-sn-ratio ] &lt;a positive number&gt;</code></p>
      <p>Set the signal/noise ratio for MS1 spectra. The default value is 3.</p>

      <p><code>-s [ --ms-two-sn-ratio ] &lt;a positive number&gt;</code></p>
      <p>Set the signal/noise ratio for MS/MS spectra. The default value is 1.</p>

      <p><code>-w [ --precursor-window ] &lt;a positive number&gt;</code></p>
      <p>Set the precursor isolation window size. The default value is 3.0 m/z.
      When the input file contains the information of precursor windows, the 
      parameter will be ignored.
      </p>

      <p><code>-n [ --env-cnn ]</code></p>
      <p>Use EnvCNN as the scoring function of isotopic envelopes. Detailed
      description of EnvCNN can be found 
      <a href="https://pubmed.ncbi.nlm.nih.gov/32356965/">here</a>.
      </p>

      <p><code>-o [ --missing-level-one ]</code></p>
      <p>Specify that the input file does not contain MS1 spectra.</p>

      <p><code>-u [ --thread-number ] &lt;a positive integer&gt;</code></p>
      <p>Number of threads used in spectral deconvolution. Default value: 1.</p>

      <p><code>-g [ --skip-html-folder ] </code></p>
      <p>Skip generation of HTML files for visualization.</p>
	  
	  <p><code>-d [ --disable-final-filtering ] </code></p>
      <p>Skip final filtering of envelopes.</p>

      <h4>Examples</h4>
      <p>
      Deconvolute a centroid data file spectra.mzML and output four files:
      spectra_ms2.msalign, spectra_ms1.feature, spectra_ms2.feature, spectra_feature.xml.
      </p>
      <p><code> topfd spectra.mzML </code></p>

      <p>
      Deconvolute two centroid data files spectra1.mzML and spectra2.mzML and 
      output four files for each input data file.
      </p>
      <p><code> topfd spectra1.mzML spectra2.mzML </code></p>

      <p>
      Deconvolute all centroid data files in the current folder. 
      </p>

      <p><code> topfd *.mzML </code></p>

      <p>
      Deconvolute a centroid data file spectra.mzML and skip final filtering and generatation of the HTML folder for visualization.  
      </p>
      <p><code> topfd -d -g spectra.mzML </code></p>

      <p>
        Deconvolute a centroid data file spectra.mzML using 4 threads and the EnvCNN scoring function.
      </p>

      <p><code> topfd -u 4 -n spectra.mzML </code></p>

      <p>
      Deconvolute a centroid data file spectra.mzML that does not contain MS1
      spectra. 
      </p>
      <p><code> topfd -o spectra.mzML </code></p>

      <p>
      Deconvolute a centroid data file spectra.mzML with a precursor isolation
      window size of 2 m/z.  
      </p>
      <p><code> topfd -w 2 spectra.mzML </code></p>

      <p>
      Deconvolute a centroid data file spectra.mzML with a signal/noise ratio 2 for MS1 spectra. 
      </p>
      <p><code> topfd -r 2 spectra.mzML </code></p>

      <p>
      Deconvolute a centroid data file spectra.mzML with the following settings: 
      the maximum charge state: 50, the maximum mass: 30,000 Dalton, and the 
      signal/noise ratio for MS/MS spectra: 2.   
      </p>
      <p><code> topfd -c 50 -m 30000 -s 2 spectra.mzML </code>
      </p>


      <h3>3 TopIndex</h3>

      <h4>3.1 Input</h4>
      <p>
      The input is a protein sequence database file in the FASTA format. 
      </p>

      <h4>3.2 Output</h4>
      <p>The output is a folder containing protein sequence index files. For example, when the input file name is
      <b>proteins.fasta</b>, the output folder is <b>proteins.fasta_idx</b>.

      <h4>3.3 Command line usage</h4>
      <p>To run TopIndex, open a terminal window and run the following command.</p>
      <p><code>topindex [options] database-file-name </code></p>

      <h4>Options</h4>
      <p><code>-h [ --help ]</code></p>
      <p>Print the help message.</p>

      <p><code>-f [ --fixed-mod ] &lt;C57|C58|a fixed modification file&gt; </code></p>
      <p>Set fixed modifications. Three available options: C57, C58, or the name of
      a text file specifying fixed modifications (see an <a
         target="_blank" href="example_files/fixed_mods.txt">example file</a>).
       When C57 is selected, carbamidomethylation on cysteine is the only fixed
       modification.  When C58 is selected, carboxymethylation on cysteine is the
       only fixed modification.</p>

      <p><code>-n [ --n-terminal-form ] &lt;a list of allowed N-terminal forms&gt;</code></p>
      <p>Set N-terminal forms of proteins. Four N-terminal forms can be selected:
      NONE, NME, NME_ACETYLATION, and M_ACETYLATION.  NONE stands for no
      modifications, NME for N-terminal methionine excision, NME_ACETYLATION for
      N-terminal acetylation after the initiator methionine is removed, and
      M_ACETYLATION for N-terminal methionine acetylation.  When multiple forms
      are allowed, they are separated by commas. Default value:
      NONE,M_ACETYLATION,NME,NME_ACETYLATION.</p>

      <p><code>-d [ --decoy ]</code></p>
      <p>Use a shuffled decoy protein database to estimate spectrum and
      proteoform-level FDRs.
      When -d is chosen, a shuffled decoy database is automatically
      generated and appended to the target database. Index files for 
      the concatenated database are generated.</p>

      <p><code>-e [ --mass-error-tolerance ] &lt;a positive integer&gt;</code></p> 
      <p>Set the error tolerance for precursor and fragment masses in ppm. Default value: 15.</p>

      <p><code>-u [ --thread-number ] &lt;a positive integer&gt;</code></p> 
      <p>Set the number of threads used in the computation. Default value: 1.
      The maximum number of threads is determined by the CPU and memory of the
      computer. About 4 GB memory is required for each
      thread. If the computer has 16 GB memory and a CPU with 8 cores, the
      maximum number of threads is 4 because about 16 GB memory is needed for 4
      threads.</p>

      <h4>Examples</h4>
      <p> Generate index files for a protein database file proteins.fasta using
      default parameters.   
      </p>
      <p><code> topindex proteins.fasta</code></p>

      <p> Generate index files for a protein database file proteins.fasta using
      carboxymethylation as the fixed modification, and N-terminal methionine
      excision and N-terminal methionine acetylation as the N-terminal forms.   
      </p>
      <p><code> topindex -f C58 -n NME,M_ACETYLATION proteins.fasta</code></p>

      <p> Generate index files for a protein database file proteins.fasta using a 
      target-decoy concatenated database, a mass error tolerance of 10 ppm, and 4 threads.</p>
      <p><code> topindex -d -e 10 -u 4 proteins.fasta</code></p>

      <h3>4 TopPIC</h3>

      <h4>4.1 Input</h4>
      <ul>
        <li>A protein database file in the FASTA format</li>
        <li>A mass spectrum data file in the msalign format</li>
        <li>A text file containing LC-MS feature information (optional)</li>
        <li>A text file of fixed PTMs (optional)</li>
        <li>A text file of PTMs for the characterization of unexpected mass shifts
        (optional)</li>
      </ul>

      <h4>4.2 Output</h4>
      <p>TopPIC outputs four tab separated value (TSV) files, two XML files, 
      and a collection of HTML files for identified
      proteoforms. For example, when the input data file is
      spectra_ms2.msalign, the output includes: 
      <ul>
        <li><b>spectra_ms2_toppic_prsm.tsv</b>: a TSV file containing 
        identified proteoform spectrum-matches (PrSMs) with an E-value or
        spectrum-level FDR cutoff. When an identified proteoform is shared by multiple 
	proteins, all the proteins are reported. </li>
        <li><b>spectra_ms2_toppic_prsm_single.tsv</b>: a TSV file containing 
        identified proteoform spectrum-matches (PrSMs) with an E-value or
        spectrum-level FDR cutoff. When an identified proteoform is shared by multiple 
	proteins, only one protein is reported.</li>
        <li><b>spectra_ms2_toppic_proteoform.tsv</b>: a TSV file containing 
        identified proteoforms with an E-value or proteoform-level FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	all the proteins are reported. </li>
        <li><b>spectra_ms2_toppic_proteoform_single.tsv</b>: a TSV file containing 
        identified proteoforms with an E-value or proteoform-level FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	only one protein is reported.
	</li>
        <li><b>spectra_ms2_toppic_proteoform.xml</b>: an XML file containing 
        identified proteoforms with the E-value or proteoform-level FDR cutoff.</li>
        <li><b>spectra_ms2_toppic_prsm.xml</b>: an XML file containing 
        all identified PrSMs without clustering and filtering.</li>
        <li><b>spectra_html/toppic_prsm_cutoff</b>: 
        a folder containing JavaScript files of identified PrSMs using the 
        E-value or spectrum-level FDR cutoff.</li>
        <li><b>spectra_html/toppic_proteoform_cutoff</b>: a folder containing
        JavaScript files of identified PrSMs using the E-value or 
        proteoform-level FDR cutoff.</li>
        <li><b>spectra_html/topmsv</b>: a folder containing
        HTML files for the visualization of identified PrSMs.</li>
      </ul>

      To browse identified
      proteins, proteoforms, and PrSMs,  
      use a chrome browser to open the file spectrum_html/topmsv/index.html.
      <b>Google Chrome</b> is recommended (Firefox and Edge are not recommended).
      </p>
      <p>When the input contains two or more data files,
      TopPIC outputs four TSV files, two XML files, and a collection of HTML files for each input file. 
      When a file name is specified for combined identifications, it combines 
      spectra and proteoforms identified from all the input files, removes redundant proteoform 
      identifications, and reports 
      four TSV files, two XML files, and a collection of HTML files for the combined results. 
      For example, when the input is spectra1_ms2.msalign and spectra2_ms2.msalign and the combined
      output file name is "combined," the output files are:
      
      <ul>
        <li><b>combined_ms2_toppic_prsm.tsv</b>: a TSV file containing 
        PrSMs identified from all the input files with an E-value or spectrum-level 
        FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	all the proteins are reported.
        </li>
        <li><b>combined_ms2_toppic_prsm_single.tsv</b>: a TSV file containing 
        PrSMs identified from all the input files with an E-value or spectrum-level 
        FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	only one protein is reported. 
	</li>
        <li><b>combined_ms2_toppic_proteoform.tsv</b>: a TSV file containing 
        proteoforms identified from all the input files with an E-value or
        proteoform-level FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	all the proteins are reported.
        </li>
        <li><b>combined_ms2_toppic_proteoform_single.tsv</b>: a TSV file containing 
        proteoforms identified from all the input files with an E-value or
        proteoform-level FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	only one protein is reported. 
	</li>
        <li><b>combined_ms2_toppic_proteoform.xml</b>: an XML file containing 
        proteoforms identified from all the input files with the E-value or
        proteoform-level FDR cutoff.</li>
        <li><b>combined_ms2_toppic_prsm.xml</b>: an XML file containing 
        all identified PrSMs without clustering and filtering.</li>
        <li><b>combined_html/toppic_prsm_cutoff</b>: a folder containing
        JavaScript files of PrSMs identified from all the input files using
        the E-value or  spectrum-level FDR cutoff.</li>
        <li><b>combined_html/toppic_proteoform_cutoff</b>: a folder containing
        JavaScript files of PrSMs identified from all the input files using the 
        E-value or proteoform-level FDR cutoff.</li>
        <li><b>combined_html/topmsv</b>: a folder containing
        HTML files for the visualization of identified PrSMs.</li>
      </ul>
       
      </p>
      <h4>4.3 Command line usage</h4>
      <p>To run TopPIC, open a terminal window and run the following command.</p>
      <p><code>toppic [options] database-file-name spectrum-file-names</code></p>

      <h4>Options</h4>
      <p><code>-h [ --help ]</code></p>
      <p>Print the help message.</p>

      <p><code>-a [ --activation ] &lt;CID|HCD|ETD|UVPD|FILE&gt;</code></p>
      <p>Set the fragmentation method(s) of MS/MS spectra. When "FILE" is
      selected, 
      the fragmentation methods of spectra are given in the input spectrum data
      file. Default value: FILE.</p>

      <p><code>-f [ --fixed-mod ] &lt;C57|C58|a fixed modification file&gt;</code></p>
      <p>Set fixed modifications. Three available options: C57, C58, or the name of
      a text file containing the information of fixed modifications (see an <a
        target="_blank" href="example_files/fixed_mods.txt">example file</a>).
      When C57 is selected, carbamidomethylation on cysteine is the only fixed
      modification.  When C58 is selected, carboxymethylation on cysteine is the
      only fixed modification.</p>

      <p><code>-n [ --n-terminal-form ] &lt;a list of allowed N-terminal forms&gt;</code></p>
      <p>Set N-terminal forms of proteins. Four N-terminal forms can be selected:
      NONE, NME, NME_ACETYLATION, and M_ACETYLATION.  NONE stands for no
      modifications, NME for N-terminal methionine excision, NME_ACETYLATION for
      N-terminal acetylation after the initiator methionine is removed, and
      M_ACETYLATION for N-terminal methionine acetylation.  When multiple forms
      are allowed, they are separated by commas. Default value:
      NONE,M_ACETYLATION,NME,NME_ACETYLATION.</p>

      <p><code>-d [ --decoy ]</code></p>
      <p>Use a shuffled decoy protein database to estimate spectrum and
      proteoform-level FDRs.
      When -d is chosen, a shuffled decoy database is automatically
      generated and appended to the target database before database search, and
      FDR rates are estimated using the target-decoy approach.</p>

      <p><code>-e [ --mass-error-tolerance ] &lt;a positive integer&gt;</code></p>
      <p>Set the error tolerance for precursor and fragment masses in part-per-million
      (ppm). Default value: 15. When the lookup table approach (<code>-l</code>) is used
      for E-value estimation, valid error tolerance values are 5, 10, and 15 ppm.</p>

	    <p><code>-p [ --proteoform-error-tolerance ]  &lt;a positive number&gt;</code></p>
      <p>Set the error tolerance for identifying PrSM clusters (in Dalton). Default value: 1.2 Dalton.</p>

      <p><code>-m [ --min-shift ]  &lt;a number&gt;</code></p>
      <p>Se the minimum value for unexpected mass shifts (in Dalton). 
      Default value: -500 Dalton.</p>

      <p><code>-M [ --max-shift ]  &lt;a number&gt;</code></p>
      <p>Set the maximum value for unexpected mass shifts (in Dalton).
      Default value: 500 Dalton.</p>

      <p><code>-s [ --num-shift ] &lt;0|1|2&gt;</code></p>
      <p>Set the maximum number of unexpected mass shifts in a PrSM. Default value: 1.</p>

      <p><code>-t [ --spectrum-cutoff-type ] &lt;EVALUE|FDR&gt;</code></p>
      <p>Set the spectrum-level cutoff type for filtering PrSMs. Default value: EVALUE.</p>

      <p><code>-v [ --spectrum-cutoff-value ] &lt;a positive number&gt;</code></p>
      <p>Set the spectrum-level cutoff value for filtering PrSMs. Default value: 0.01.</p>

      <p><code>-T [ --proteoform-cutoff-type ] &lt;EVALUE|FDR&gt;</code></p>
      <p>Set the proteoform-level cutoff type for filtering proteoforms and
      PrSMs. Default value: EVALUE.</p>

      <p><code>-V [ --proteoform-cutoff-value ] &lt;a positive number&gt;</code></p>
      <p>Set the proteoform-level cutoff value for filtering proteoforms and
      PrSMs. Default value: 0.01.</p>

      <p><code>-l [ --lookup-table ]</code></p>
      <p>Use a lookup table method for computing E-values.
      It is faster than the default generating function approach, but it
      may reduce the number of identifications.</p>

      <p><code>-r [ --num-combined-spectra ] &lt;a positive integer&gt;</code></p>
      <p>Set the number of combined spectra. The parameter is set to 2 (or 3)
      for combining spectral pairs (or triplets) generated by the alternating
      fragmentation mode. Default value: 1.</p>

      <p><code>-i [ --mod-file-name ] &lt;a common modification file&gt; </code></p>
      <p>Specify a text file containing a list of common PTMs for proteoform
      characterization. The PTMs are used to identify and localize PTMs in
      reported PrSMs with unknown mass shifts. See an 
      <a target="_blank" href="example_files/common_mods.txt">example file</a>.</p>

      <p><code>-H [ --miscore-threshold ] &lt;a number between 0 and 1&gt;</code></p>
      <p>Set the score threshold (MIScore) for filtering results of PTM
      characterization. Default value: 0.15.</p>

      <p><code>-u [ --thread-number ] &lt;a positive number&gt;</code></p>
      <p>Set the number of threads used in the computation. Default value: 1.
      The maximum number of threads is determined by the CPU and memory of the
      computer. About 4 GB memory is required for each thread. If the computer
      has 16 GB memory and a CPU with 8 cores, the maximum number of threads is
      4 because about 16 GB memory is needed for 4 threads.</p>

      <p><code>-x [ --no-topfd-feature ]</code></p>
      <p>Specify that there are no TopFD feature files for proteoform identification.</p>

      <p><code>-c [ --combined-file-name ] &lt;a filename&gt;</code></p>
      <p>Specify an output file name for combined identifications when the input
      consists of multiple data files.</p>

      <p><code>-k [ --keep ]</code></p>  
      <p>Keep intermediate files. 
      </p> 

      <p><code>-K [ --keep-decoy-ids ]</code></p>  
      <p>Keep decoy identifications.
      </p> 

      <p><code>-g [ --skip-html-folder ] </code></p>
      <p>Skip the generation of HTML files for visualization.</p>

      <h4>Examples</h4>
      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta with a feature file
      spectra_ms2.feature (reported by TopFD). 
      The user does not need to specify the feature file name. TopPIC will automatically 
      obtain the feature file name from the spectrum file name spectra_ms2.msalign.</p>
      <p><code> toppic proteins.fasta spectra_ms2.msalign </code></p>
      
      <p> Search two deconvoluted MS/MS spectrum files spectra1_ms2.msalign and 
      spectra2_ms2.msalign against a protein database file proteins.fasta with feature files. 
      In addition, all identifications are combined and reported using a file name "combined." 
      </p>
      <p><code> toppic -c combined proteins.fasta spectra1_ms2.msalign spectra2_ms2.msalign </code></p>

      <p> Search all deconvoluted MS/MS spectrum files in the current folder
      against a protein database file proteins.fasta with feature files. 
      </p>
      <p><code> toppic proteins.fasta *_ms2.msalign </code></p>
      
      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta without feature files.</p>
      <p><code> toppic -x proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta with a feature file
      and a fixed modification: carbamidomethylation on cysteine. </p>
      <p><code> toppic -f C57 proteins.fasta spectra_ms2.msalign </code></p>
      
      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file. 
      In an identified proteoform, at most 2 mass shifts are
      allowed and the maximum allowed mass shift value is 10,000 Dalton. </p>
      <p><code> toppic -s 2 -M 10000 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file.
      The error tolerance for precursor and fragment masses is
      5 ppm. </p> 
      <p><code> toppic -e 5 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file.
      Use the target decoy approach to compute spectrum level
      and proteoform-level FDRs, filter identified proteoform
      spectrum-matches by a 5% spectrum level FDR, and filter identified
      proteoforms by a 5% proteoform-level FDR. 
      <p><code> toppic -d -t FDR -v 0.05 -T FDR -V 0.05 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign with
      alternating CID, HCD, and ETD spectra against a protein database file
      proteins.fasta with a feature file. Combine alternating
      CID, HCD, and ETD spectra to increase proteoform coverage. 
      <p><code> toppic -r 3 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file.
      After proteoforms with unexpected mass shifts are
      identified, TopPIC matches the mass shifts to four common PTMs: acetylation,
      phosphorylation, oxidation and methylation, and uses an MIScore cutoff 0.1
      to filter reported PTM sites. The modification file common_mods.txt can be 
      found <a target="_blank" href="example_files/common_mods.txt">here</a>.</p>
      <p><code> toppic -i common_mods.txt -H 0.1 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file.
      Use 6 CPU threads to speed up the computation. 
      <b><font color=#FF0000>About 24 GB memory is required for 6 threads (4 GB
          per thread). If the computer lacks enough memory, TopPIC may crash.</font></b  
      </p> 
      <p><code> toppic -u 6 proteins.fasta spectra_ms2.msalign </code></p>



      <h3>5 TopMG</h3>

      <h4>5.1 Input</h4>
      <ul>
        <li>A protein database file in the FASTA format</li>
        <li>A mass spectrum data file in the msalign format</li>
        <li>A text file of variable PTMs</li>
        <li>A text file of fixed PTMs (optional)</li>
        <li>A text file containing LC-MS feature information (optional)</li>
      </ul>

        
		<h4>5.2 Output</h4>
      <p>TopMG outputs two TSV files, an XML file, and a collection of HTML files for identified
      proteoforms. For example, when the input mass spectrum data file is
      spectra_ms2.msalign, the output includes: 
      <ul>
        <li><b>spectra_ms2_topmg_prsm.tsv</b>: a TSV file containing 
        identified PrSMs with an E-value or spectrum-level 
        FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	all the proteins are reported. 
	</li>
        <li><b>spectra_ms2_topmg_prsm_single.tsv</b>: a TSV file containing 
        identified PrSMs with an E-value or spectrum-level 
        FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	only one protein is reported. 
	</li>
        <li><b>spectra_ms2_topmg_proteoform.tsv</b>: a TSV file containing 
        identified proteoforms with an E-value or proteoform-level FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	all the proteins are reported. 
        </li>
        <li><b>spectra_ms2_topmg_proteoform_single.tsv</b>: a TSV file containing 
        identified proteoforms with an E-value or proteoform-level FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	only one protein is reported. 
        </li>
        <li><b>spectra_ms2_topmg_proteoform.xml</b>: an XML file containing 
        identified proteoforms with the E-value or proteoform-level FDR cutoff.</li>
        <li><b>spectra_ms2_topmg_prsm.xml</b>: an XML file containing all identified PrSMs without clustering and filtering.</li>
        <li><b>spectra_html/topmg_prsm_cutoff</b>: a folder containing JavaScript files of 
          identified PrSMs using the E-value or spectrum-level FDR cutoff.</li>
        <li><b>spectra_html/topmg_proteoform_cutoff</b>: a folder containing
        JavaScript files of identified PrSMs using the E-value or  
        proteoform-level cutoff.</li>
        <li><b>spectra_html/topmsv</b>: a folder containing
        HTML files for the visualization of identified PrSMs.</li>
      </ul>

      To browse identified
      proteins, proteoforms, and PrSMs,  
      use a chrome browser to open the file spectra_html/topmsv/index.html.
      <b>Google Chrome</b> is recommended (Firefox and Edge are not recommended).
      </p>

      <p>When the input contains two or more spectrum files, TopMG outputs two
      TSV files, an XML file, and a collection of HTML files for each input file. 
      When a file name is specified for combined identifications, it combines 
      spectra and proteoforms identified from all the input files, removes redundant proteoform 
      identifications, and reports 
      two TSV files, an XML file, and a collection of HTML files for the combined results. 
      
      For example, when the input is spectra1_ms2.msalign and spectra2_ms2.msalign and the combined
      output file name is "combined," the output files are:      

      <ul>
        <li><b>combined_ms2_topmg_prsm.tsv</b>: a TSV file containing 
        PrSMs identified from all the input files with an E-value or spectrum-level
        FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	all the proteins are reported. 
	</li>
        <li><b>combined_ms2_topmg_prsm_single.tsv</b>: a TSV file containing 
          PrSMs identified from all the input files with an E-value or spectrum-level
          FDR cutoff. 
	  When an identified proteoform is shared by multiple proteins, 
	  only one protein is reported. 
	</li>
        <li><b>combined_ms2_topmg_proteoform.tsv</b>: a TSV file containing 
        proteoforms identified from all the input files with an E-value or
        proteoform-level FDR cutoff. 
	When an identified proteoform is shared by multiple proteins, 
	all the proteins are reported. 
	</li>
        <li><b>combined_ms2_topmg_proteoform_single.tsv</b>: a TSV file containing 
          proteoforms identified from all the input files with an E-value or
          proteoform-level FDR cutoff. 
	  When an identified proteoform is shared by multiple proteins, 
	  only one protein is reported. 
	</li>
        <li><b>combined_ms2_topmg_proteoform.xml</b>: an XML file containing 
        proteoforms identified from all the input files with the E-value or
        proteoform-level FDR cutoff.</li>
        <li><b>combined_ms2_topmg_prsm.xml</b>: an XML file containing all identified PrSMs without clustering and filtering.</li>
        <li><b>combined_html/topmg_prsm_cutoff</b>: a folder containing
          JavaScript files of PrSMs identified from all the input files using
          the E-value or spectrum-level FDR cutoff.</li>
        <li><b>combined_html/topmg_proteoform_cutoff</b>: a folder containing
        JavaScript files of PrSMs identified from all the input files using the 
        E-value or proteoform-level cutoff.</li>
        <li><b>combined_html/topmsv</b>: a folder containing
        HTML files for the visualization of identified PrSMs.</li>
      </ul>      
      
      <h4>5.3 Command line usage</h4>
      <p>To run TopMG, open a terminal window and run the following command.</p>
      <p><code>topmg [options] database-file-name spectrum-file-names</code></p>

      <h4>Options</h4>
      <p><code>-h [ --help ]</code></p>
      <p>Print the help message.</p>

      <p><code>-a [ --activation ] &lt;CID|HCD|ETD|UVPD|FILE&gt;</code></p>
      <p>Fragmentation method of MS/MS spectra. When FILE is used, 
        fragmentation methods of spectra are given in the input spectral data file. 
        Default value: FILE.</p>

      <p><code>-f [ --fixed-mod ] &lt;C57|C58|a fixed modification file&gt;</code></p>
      <p>Set fixed modifications. Three available options: C57, C58, or the name of
      a text file specifying fixed modifications (see an <a
        target="_blank" href="example_files/fixed_mods.txt">example file</a>).
      When C57 is selected, carbamidomethylation on cysteine is the only fixed
      modification.  When C58 is selected, carboxymethylation on cysteine is the
      only fixed modification.</p>

      <p><code>-n [ --n-terminal-form ] &lt;a list of allowed N-terminal forms&gt;</code></p>
      <p>Set N-terminal forms of proteins. Four N-terminal forms can be selected:
      NONE, NME, NME_ACETYLATION, and M_ACETYLATION.  NONE stands for no
      modifications, NME for N-terminal methionine excision, NME_ACETYLATION for
      N-terminal acetylation after the initiator methionine is removed, and
      M_ACETYLATION for N-terminal methionine acetylation.  When multiple forms
      are allowed, they are separated by commas. Default value:
      NONE,M_ACETYLATION,NME,NME_ACETYLATION.</p>

      <p><code>-d [ --decoy ]</code></p>
      <p>Use a shuffled decoy protein database to estimate spectrum and
      proteoform level FDRs.
      When -d is chosen, a shuffled decoy database is automatically
      generated and appended to the target database before database search, and
      FDR rates are estimated using the target-decoy approach.</p>

      <p><code>-e [ --mass-error-tolerance ] &lt;a positive integer&gt;</code></p>
      <p>Set the error tolerance for precursor and fragment masses in ppm.
      Default value: 15 ppm.</p>

      <p><code>-p [ --proteoform-error-tolerance ] &lt;a positive number&gt;</code></p>
      <p>Set the error tolerance for identifying PrSM clusters (in Dalton). Default value: 1.2 Dalton.</p>

      <p><code>-M [ --max-shift ]  &lt;a number&gt;</code></p>
      <p>Set the maximum absolute value for unexpected mass shifts (in Dalton).
      Default value: 500 Dalton.</p>

      <p><code>-t [ --spectrum-cutoff-type ] &lt;EVALUE|FDR&gt;</code></p>
      <p>Set the spectrum level cutoff type for filtering PrSMs. Default value: EVALUE.</p>

      <p><code>-v [ --spectrum-cutoff-value ] &lt;a positive number&gt;</code></p>
      <p>Set the spectrum level cutoff value for filtering PrSMs. Default value: 0.01.</p>

      <p><code>-T [ --proteoform-cutoff-type ] &lt;EVALUE|FDR&gt;</code></p>
      <p>Set the proteoform level cutoff type for filtering proteoforms and
      PrSMs. Default value: EVALUE.</p>

      <p><code>-V [ --proteoform-cutoff-value ] &lt;a positive number&gt;</code></p>
      <p>Set the proteoform level cutoff value for filtering proteoforms and
      PrSMs. Default value: 0.01.</p>

      <p><code>-i [ --mod-file-name ] &lt;a modification file&gt; </code></p>
      <p>Specify a text file of variable PTMs. 
        See an <a target="_blank" href="example_files/variable_mods.txt" >example file</a>.</p>

      <p><code>-u [ --thread-number ] &lt;a positive number&gt;</code></p>
      <p>Set the number of threads used in the computation. Default value: 1.
			The maximum number of threads is determined by the CPU and memory 
			of the computer. About 4 GB memory is required for each thread. 
			If the computer has 16 GB memory and a CPU with 8 cores, the maximum number 
			of threads is 4 because 16 GB memory is required for 4 threads.      
      </p>

      <p><code>-x [ --no-topfd-feature ]</code></p>
      <p>Specify that there are no TopFD feature files for proteoform identification.</p>

      <p><code>-D [ --use-asf-diagonal ]</code></p>
      <p>Use the ASF-DIAGONAL method for protein sequence filtering. 
      The default filtering method is ASF-RESTRICT. When -D is selected, both ASF-RESTRICT 
      and ASF-DIAGONAL will be used. The combined approach may identify more PrSMs, 
      but it is much slower than using ASF-RESTRICT only. See 
      <a target="_blank" href="https://www.ncbi.nlm.nih.gov/pubmed/29327814">this paper</a>    
      for more details.</p>

      <p><code>-P [ --var-ptm ] &lt;a positive number&gt;</code></p>
      <p>Set the maximum number of variable PTM sites in a proteoform. Default value: 5.</p>

      <p><code>-s [ --num-shift  &lt;0|1|2&gt;</code></p>
      <p>Set the maximum number of unexpected mass shifts in a proteoform. Default value: 0.</p>

      <p><code>-w [ --whole-protein-only ] </code></p>
      <p>Report only proteoforms of whole protein sequences.  
      </p>

      <p><code>-c [ --combined-file-name ] &lt;a filename&gt;</code></p>
      <p>Specify an output file name for combined identifications when the input consists of multiple spectrum files. 
      </p>

      <p><code>-k [ --keep ]</code></p>  
      <p>Keep intermediate files. 
      </p>

      <p><code>-K [ --keep-decoy-ids ]</code></p>  
      <p>Keep decoy identifications.
      </p> 

      <p><code>-g [ --skip-html-folder ] </code></p>
      <p>Skip the generation of HTML files for visualization.</p>

      <h4>Advanced options</h4>
      <p><code>-j [ --proteo-graph-dis ] &lt;a positive number&gt;</code></p>
      <p>Set the length of the largest gap in constructing proteoform graphs. Default value: 40. 
      See 
      <a target="_blank" href="https://www.ncbi.nlm.nih.gov/pubmed/28453668">this paper</a>    
      for more details.</p>
      </p>

      <p><code>-G [ --var-ptm-in-gap ] &lt;a positive number&gt;</code></p>
      <p>Set the maximum number of variable PTM sites in a gap in a proteoform graph. Default value: 5.
      See 
      <a target="_blank" href="https://www.ncbi.nlm.nih.gov/pubmed/28453668">this paper</a>    
      for more details.</p>
      </p>

      <h4>Examples</h4>
      <p> To use the following examples, the current folder needs to contain a
      variable modification file variable_mods.txt. 
      (See an  <a target="_blank" href="example_files/variable_mods.txt">example</a>.)
      </p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta with a feature file
      spectra_ms2.feature. 
      The user does not need to specify the feature file name. TopMG will automatically 
      obtain the feature file name from the spectrum file name spectra_ms2.msalign.</p>
      <p><code> topmg -i variable_mods.txt proteins.fasta spectra_ms2.msalign </code></p>
      
      <p> Search two deconvoluted MS/MS spectrum files spectra1_ms2.msalign and 
      spectra2_ms2.msalign against a protein database file proteins.fasta with feature files.
      In addition, all identifications are combined and reported using a file name "combined." 
      </p>
      <p><code> topmg -i variable_mods.txt -c combined proteins.fasta spectra1_ms2.msalign spectra2_ms2.msalign </code></p>

      <p> Search all deconvoluted MS/MS spectrum files in the current folder
      against a protein database file proteins.fasta with feature files. 
      </p>
      <p><code> topmg -i variable_mods.txt proteins.fasta *_ms2.msalign </code></p>
      
      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta without feature files.</p>
      <p><code> topmg -i variable_mods.txt -x proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against a
      protein database file proteins.fasta with a feature file
      and a fixed modification: carbamidomethylation on cysteine. </p>
      <p><code> topmg -i variable_mods.txt -f C57 proteins.fasta spectra_ms2.msalign </code></p>
      
      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file.
      In an identified proteoform, at most 1 unexpected mass shift and 
      4 variable PTMs are allowed and the maximum value for unexpected mass shifts is 10,000 Dalton. </p>
      <p><code> topmg -i variable_mods.txt -P 4 -s 1 -M 10000 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file.
      The error tolerance for precursor and fragment masses is
      5 ppm. </p> 
      <p><code> topmg -i variable_mods.txt -e 5 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file.
      Use the target decoy approach to compute spectrum level
      and proteoform level FDRs, filter identified proteoform
      spectrum-matches by a 5% spectrum-level FDR, and filter identified
      proteoforms by a 5% proteoform-level FDR.
      </p>
      <p><code> topmg -i variable_mods.txt -d -t FDR -v 0.05 -T FDR -V 0.05 proteins.fasta spectra_ms2.msalign </code></p>

      <p> Search a deconvoluted MS/MS spectrum file spectra_ms2.msalign against
      a protein database file proteins.fasta with a feature file.
      Use 6 CPU threads to speed up the computation. <b><font
            color=#FF0000>About 24 GB memory is required for 6 threads (4 GB per
            thread). If the computer lacks enough memory, TopMG may crash.</font></b 
      </p> 
      <p><code> topmg -i variable_mods.txt -u 6 proteins.fasta spectra_ms2.msalign </code></p>      

      <h3>6 TopDiff</h3>

      <h4>6.1 Input</h4>
      <ul>
        <li>Proteoform identification files of the mass spectrum data files 
        in the XML format, e.g., spectra_ms2_toppic_proteoform.xml </li>
      </ul>


      <h4>6.2 Output</h4>
      <p>TopDiff outputs a TSV file containing proteoform identifications
      and their abundances in the input mass spectrum data. 
      The default output file name is sample_diff.tsv.
      
      <h4>6.3 Command line usage</h4>
      <p>To run TopDiff, open a terminal window and run the following command.</p>
      <p><code>topdiff [options] spectrum-file-names</code></p>

      <h4>Options</h4>
      <p><code>-h [ --help ]</code></p>
      <p>Print the help message.</p>

      <p><code>-e [ --error-tolerance ] &lt;a positive number&gt;</code></p>
      <p>Set the error tolerance for mapping identified proteoforms across
      multiple samples (in Dalton). Default value: 1.2 Dalton.</p>

      <p><code>-t [ --tool-name ] &lt;toppic|topmg&gt;</code></p>
      <p>
      Specify the name of the database search tool: toppic or topmg. Default: toppic.
      </p>

      <p><code>-o [ --output ] &lt;a file name&gt;</code></p>
      <p>
      Specify the output file name. Default: sample_diff.tsv.
      </p>

      <h4>Examples</h4>
      <p> Compare proteoform abundances using TopPIC identifications of two spectrum files
      spectra1_ms2.msalign and spectra2_ms2.msalign. 
      </p>
      <p><code> topdiff spectra1_ms2.msalign spectra2_ms2.msalign </code></p>

      <p> Compare proteoform abundances using TopMG identifications of two spectrum files
      spectra1_ms2.msalign and spectra2_ms2.msalign.
      </p>
      <p><code> topdiff -t topmg spectra1_ms2.msalign spectra2_ms2.msalign </code></p>

      <!--
      <h3>7 TopMerge</h3>

      <h4>7.1 Input</h4>
      <ul>
        <li>A protein database file in the FASTA format</li>
        <li>Several mass spectrum data files in the msalign format</li>
        <li>Proteoform identification files of the mass spectrum data files 
          in the XML format, e.g., spectra_ms2_toppic_proteoform.xml </li>
        <li>Feature files containing LC-MS features, and MS/MS scan IDs and their
          corresponding LC-MS feature IDs. format, e.g., spectra_ms1.feature, spectra_ms2.feature </li>
      </ul>

      <h4>7.2 Output</h4>
      <p>TopMerge combines proteoform identifications and features from all input files to
        create a single output file for each file type (tsv, xml, feature).
      
      <h4>6.3 Command line usage</h4>
      <p>To run TopMerge, open a terminal window and run the following command.</p>
      <p><code>topmerge [options] database-file-name spectrum-file-name-1 spectrum-file-name-2 spectrum-file-name-3-or-more</code></p>

      <h4>Options</h4>
      <p><code>-h [ --help ]</code></p>
      <p>Print the help message.</p>

      <p><code>-d [ --decoy ]</code></p>
      <p>Use a shuffled decoy protein database to estimate spectrum and
      proteoform level FDRs.</p>

      <p><code>-e [ --mass-error-tolerance ] &lt;a positive integer&gt;</code></p>
      <p>Set the error tolerance for precursor and fragment masses in part-per-million
      (ppm). Default value: 15. When the lookup table approach (<code>-l</code>) is used
      for E-value estimation, valid error tolerance values are 5, 10, and 15 ppm.</p>

	    <p><code>-p [ --proteoform-error-tolerance ]  &lt;a positive number&gt;</code></p>
      <p>Set the error tolerance for identifying PrSM clusters (in Dalton). Default value: 1.2 Dalton.</p>

      <p><code>-M [ --max-shift ]  &lt;a number&gt;</code></p>
      <p>Set the maximum value for unexpected mass shifts (in Dalton).
      Default value: 500 Dalton.</p>

      <p><code>-m [ --min-shift ]  &lt;a number&gt;</code></p>
      <p>Se the minimum value for unexpected mass shifts (in Dalton). 
      Default value: -500 Dalton.</p>

      <p><code>-t [ --spectrum-cutoff-type ] &lt;EVALUE|FDR&gt;</code></p>
      <p>Set the spectrum-level cutoff type for filtering PrSMs. Default value: EVALUE.</p>

      <p><code>-v [ --spectrum-cutoff-value ] &lt;a positive number&gt;</code></p>
      <p>Set the spectrum-level cutoff value for filtering PrSMs. Default value: 0.01.</p>

      <p><code>-T [ --proteoform-cutoff-type ] &lt;EVALUE|FDR&gt;</code></p>
      <p>Set the proteoform-level cutoff type for filtering proteoforms and
      PrSMs. Default value: EVALUE.</p>

      <p><code>-V [ --proteoform-cutoff-value ] &lt;a positive number&gt;</code></p>
      <p>Set the proteoform-level cutoff value for filtering proteoforms and
      PrSMs. Default value: 0.01.</p>

      <p><code>-i [ --mod-file-name ] &lt;a common modification file&gt; </code></p>
      <p>Specify a text file containing a list of common PTMs for proteoform
      characterization. The PTMs are used to identify and localize PTMs in
      reported PrSMs with unknown mass shifts. See an 
      <a target="_blank" href="example_files/common_mods.txt">example file</a>.</p>

      <p><code>-H [ --miscore-threshold ] &lt;a number between 0 and 1&gt;</code></p>
      <p>Set the score threshold (MIScore) for filtering results of PTM
      characterization. Default value: 0.45.</p>

      <p><code>-x [ --no-topfd-feature ]</code></p>
      <p>Specify that there are no TopFD feature files for proteoform identification.</p>

      <p><code>-c [ --combined-file-name ] &lt;a filename&gt;</code></p>
      <p>Specify an output file name for combined identifications when the input
      consists of multiple data files.</p>

      <p><code>-k [ --keep ]</code></p>  
      <p>Keep intermediate files. 
      </p> 

      <p><code>-K [ --keep-decoy-ids ]</code></p>  
      <p>Keep decoy identifications.
      </p> 

      <p><code>-g [ --skip-html-folder ] </code></p>
      <p>Skip generating the folder for spectral visualization.</p>


      <h4>Examples</h4>
      <p> Merge identification results from two spectrum files spectra1_ms2.msalign and spectra2_ms2.msalign. 
      The protein sequence database file name is proteins.fasta.  
      </p>
      <p><code> topmerge proteins.fasta spectra1_ms2.msalign spectra2_ms2.msalign </code></p>

      <p> Merge identification results from two spectrum files
      spectra1_ms2.msalign and spectra2_ms2.msalign. The protein sequence
      database file name is proteins.fasta. Use the target decoy approach to compute spectrum level
      and proteoform-level FDRs, filter identified proteoform
      spectrum-matches by a 5% spectrum level FDR, and filter identified
      proteoforms by a 5% proteoform-level FDR.</p>
      </p>
      <p><code> topmerge -d -t FDR -v 0.05 -T FDR -V 0.05 proteins.fasta spectra1_ms2.msalign spectra2_ms2.msalign </code></p>

      <p> Merge identification results from two spectrum files
      spectra1_ms2.msalign and spectra2_ms2.msalign. The protein sequence
      database file name is proteins.fasta. Set the output file name as "merged." Skip generation of html folders . </p>
      </p>
      <p><code> topmerge -c merged -k proteins.fasta spectra1_ms2.msalign spectra2_ms2.msalign </code></p>
      -->
      
    </div>
  </div>
</div>
<script src="assets/js/jquery-1.7.1.min.js"></script>

</body>
</html>
