<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <title>TopPIC Suite</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="">
  <meta name="author" content="xiaowen">

  <link href="assets/css/bootstrap.css" rel="stylesheet">
  <link href="style.css" rel="stylesheet">
  <link href="assets/css/bootstrap-responsive.css" rel="stylesheet">
  <link rel="shortcut icon" href="toppic.ico">
</head>

<body style="overflow-y:scroll">

  <div class="container">

    <div class="masthead">
      <h2 class="muted">
        <font color=#0066DD>TopPIC Suite</font>
      </h2>
      <div class="navbar">
        <div class="navbar-inner">
          <div class="container">
            <ul class="nav">
              <li><a href="index.html">Summary</a></li>
              <li><a href="register.html">Downloads</a></li>
              <li><a href="manual.html">Manual</a></li>
              <li><a href="tutorial.html">Tutorial</a></li>
              <li><a href="publications.html">Publications</a></li>
              <li><a href="contact.html">Contact</a></li>
            </ul>
          </div>
        </div>
      </div><!-- /.navbar -->
    </div><!-- /.masthead -->

    <div class="row-fluid" style="font-size:16px">

      <div class="span-12">
        <p>
          In addition to this tutorial, you can find video tutorials on TopPIC
          Suite (<a href="https://www.youtube.com/watch?v=l6KRtTrXMiw">link</a>)
          and the interpretation of TopPIC and TopMG identifications (<a
            href="https://www.youtube.com/watch?v=r4ihSms9DJAi">link</a>). We thank
          Dr. David Tabb for making these video tutorials.
        </p>


        <h3>1 Overview</h3>
        <p>
          In this tutorial, we use TopPIC Suite to analyze two top-down LC-MS/MS data files
          on a computer with a Windows Operating System.
          <b>Annotated proteoform spectrum matches (PrSMs) identified by TopPIC from the data files can be browsed
            <a href="data_files/st_1_html/topmsv/index.html">here</a></b>.
        </p>

        <hr>

        <h3>2 Folders</h3>
        <p>Create the folders below for software packages and data sets
          used in this tutorial.
        </p>
        <ol class="textList">
          <li>Create a new folder named <code>toppic_tutorial</code> on the <code>C:</code> drive of your system.
          <li>Create a new subfolder named <code>toppic</code> in the folder <code>C:\toppic_tutorial\</code> for
            the software TopPIC suite.
          <li>Create a new subfolder named <code>tutorial_1</code> in the folder <code>C:\toppic_tutorial\</code>.
          <li>Create a new subfolder named <code>tutorial_2</code> in the folder <code>C:\toppic_tutorial\</code>.
          <li>Create a new subfolder named <code>tutorial_3</code> in the folder <code>C:\toppic_tutorial\</code>.
          <li>Create a new subfolder named <code>tutorial_4</code> in the folder <code>C:\toppic_tutorial\</code>.
        </ol>
        <p>The resulting folder structure is shown in the screenshot below.</p>
        <br>
        <p>
          <img src="./images/folder_1.png" height="200" width="600" style="border:
      #00008B 1px solid;" />
        </p>

        <hr>

        <h3>3 Software tools</h3>

        <h4 class="sectionTitle">3.1 Msconvert</h4>
        <p>
          Msconvert is a software tool in <B>ProteoWizard</B> that converts raw
          files into various spectrum file formats. Follow the steps below to download ProteoWizard:
        </p>
        <p>
        <ol class="textList">
          <li>Go to the link <a href="http://proteowizard.sourceforge.net/download.html">ProteoWizard</a>.</li>
          <li>Choose the type 'Windows 64-bit installer' for end users and download ProteoWizard.</li>
          <li>Double click the downloaded file pwiz-setup-3.0-x86-64.msi to install it. </li>
        </ol>
        </p>

        <h4 class="sectionTitle">3.2 TopPIC suite</h4>
        <p>
        <ol class="textList">
          <li>Go to the <a href="./register.html">download webpage</a> of TopPIC suite.</li>
          <li>Choose the download type "Windows 64-bit zip file", fill out the
            registration form, and click "I accept license agreement and download
            TopPIC Suite"
            to download it.</li>
          <li>Save it to the folder <code>C:\toppic_tutorial\toppic\</code>.</li>
          <li>Extract all the files of the downloaded zip file to the folder
            <code>C:\toppic_tutorial\toppic\</code>.
          </li>
          </p>
          <p><img src="./images/toppic_1.png" height="200" width="600" style="border: #00008B 1px solid;" />
          </p>
        </ol>

        <hr>

        <h3>4 Tutorial 1</h3>
        In this tutorial, we will use TopIndex, TopFD and TopPIC to analyze a
        top-down MS/MS data set of Salmonella typhimurium for proteoform
        identification.
        <h4 class="sectionTitle">4.1 Top-down MS/MS Dataset</h4>
        <p>
          In the MS experiment, the protein extract of S. typhimurium was reduced
          with dithiothreitol and alkylated with iodoacetamide. The protein
          mixture was first separated by gas-phase fractionation, resulting in 7
          fractions. Each fraction was separated by an HPLC system coupled with an
          LTQ-Orbitrap mass spectrometer (Thermo Fisher Scientific). MS and MS/MS
          spectra were collected at a resolution of 60,000 and 30,000,
          respectively. In this tutorial, we use only the data files of two
          fractions (st_1.raw and st_2.raw).
        </p>
        <p>Click <a href="./data_files/tutorial-1-raw.zip">here</a>
          to download the data set, save it in the folder <code>C:\toppic_tutorial\tutorial_1\</code>, and unzip it in
          the same folder.</p>

        <h4 class="sectionTitle">4.2 Protein sequence database</h4>
        <p>A S. typhimurium <a href="https://www.uniprot.org/proteomes/UP000001014">proteome database</a>
          of 4,533 proteins was downloaded from the UniProt database.
        </p>

        <p>Click <a href="filedownload.php?filename=./data_files/uniprot-st.fasta">here</a>
          to download the protein database and save it in the folder
          <code>C:\toppic_tutorial\tutorial_1\</code>.
        </p>

        <p>The folder <code>C:\toppic_tutorial\tutorial_1\</code> is shown in the screenshot below.</p>
        <br>
        <img src="./images/folder_2.png" height="200" width="600" style="border: #00008B 1px solid;" /></p>
        </p>

        <h4 class="sectionTitle">4.2.1 Index file generation</h4>
        <p>We use <b>TopIndex</b> to generate index files from the protein
          database. They will speed up database search of TopPIC and TopMG.
          This step is optional. Skipping index generation only slows the analysis
          of section 4.5 for database search.
          <b>While TopIndex supports multithreading, users with a spinning hard
            disk would experience faster speed when using only one thread instead of
            multple threads. TopIndex generates very large index files. For example,
            index files generated for the targe-decoy concatenated UniProt human
            proteome database are about 240 GB.
            To achieve high speed index
            generation, we suggest that a computer with at least 1 TB SSD (Solid State
            Drive) should be used.</b>
        </p>
        <ol class="textList">
          <li> Double click the executable file <code>topindex_gui.exe</code> in the folder
            <code>C:\toppic_tutorial\toppic</code>.
          </li>
          <li> Add the file <code>C:\toppic_tutorial\tutorial_1\uniprot-st.fasta</code>.</li>
          <li> Select <code>Carbamidomethylation on cysteine</code> as the fixed modification. </li>
          <li> Check the checkbox <code>Decoy database</code>. </li>
          <li> Click to the button "Start" to generate index files.</li>
        </ol>
        <p>
          The screenshot of <code>topindex_gui</code> is shown below.
        </p>
        <p><img src="./images/topindex_1.png" height="400" width="600" style="border: #00008B 1px solid;" />
        </p>
        <p>
          TopIndex generates a folder
          <code>C:\toppic_tutorial\tutorial_1\uniprot-st.fasta_idx</code>
          containing index files.
        </p>

        <p> In the analysis, carbamidomethylation is selected as the fixed modification because
          proteins were reduced with dithiothreitol and alkylated with
          iodoacetamide before the MS experiment. When proteins are not reduced,
          no fixed modification should be selected.
        </p>

        <h4 class="sectionTitle">4.3 File format conversion</h4>

        <p>We use <b>MSConvertGUI</b> to convert the raw files st_1.raw and
          st_2.raw to mzML files.</p>
        <ol class="textList">
          <li> Search "msconvert" in the search box on the task bar of Windows 10,
            and run the desktop app "MSConvert."</li>
          <li> Add the files <code>C:\toppic_tutorial\tutorial_1\st_1.raw</code>
            and <code>C:\toppic_tutorial\tutorial_1\st_2.raw</code> as input
            files.</li>
          <li> Add the filter "peakPeaking vendor msLevel=1-" <b>
              <font color=#FF0000>(important)</font>
            </b>.</li>
          <li> Click to the button "Start" to perform file format conversion.</li>
        </ol>
        <p>
          The screenshot of MSConvertGUI is shown below.
        </p>
        <p><img src="./images/msconvert_1.png" height="400" width="600" style="border: #00008B 1px solid;" />
        </p>

        <p>
          In the above file format conversion, the peak picking filter (step 3) is used to
          generate centroid, not profile, mzML data files, which are required by the
          spectral deconvolution tool TopFD.
        </p>

        <p>
          The resulting mzML files are
        <pre>C:\toppic_tutorial\tutorial_1\st_1.mzML</pre> and
        <pre>C:\toppic_tutorial\tutorial_1\st_2.mzML</pre>
        The sizes of the two files are about 41
        MB and 47 MB, respectively. They can be downloaded <a href="./data_files/tutorial-1-mzml.zip">here</a>.
        The
        running time for the file format conversion is less than one minute.
        </p>

        <h4 class="sectionTitle">4.4 Mass spectral deconvolution</h4>

        <p>We use <b>topfd_gui</b> for top-down mass spectral deconvolution.</p>
        <ol class="textList">
          <li> Double click the executable file <code>topfd_gui.exe</code> in the folder
            <code>C:\toppic_tutorial\toppic</code>.
          </li>
          <li> Add the file <code>C:\toppic_tutorial\tutorial_1\st_1.mzML</code>
            and <code>C:\toppic_tutorial\tutorial_1\st_2.mzML</code> as input files.</li>
          <li> Click to the button "Start" to deconvolute the file.</li>
        </ol>
        <p>
          The screenshot of <code>topfd_gui</code> is shown below.
        </p>

        <p><img src="./images/topfd_1.png" height="400" width="600" style="border: #00008B 1px solid;" />
        </p>

        <p>TopFD reports ten text files and two folders.</p>

        <ol class="textList">
          <li>Two msalign files containing deconvoluted MS1 spectra:
            <pre>C:\toppic_tutorial\tutorial_1\st_1_ms1.msalign</pre>
            <pre>C:\toppic_tutorial\tutorial_1\st_2_ms1.msalign</pre>
          </li>
          <li>Two msalign files containing deconvoluted MS/MS spectra:
            <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2.msalign</pre>
            <pre>C:\toppic_tutorial\tutorial_1\st_2_ms2.msalign</pre>
          </li>
          <li>Four text files containing LC-MS features:
            <pre>C:\toppic_tutorial\tutorial_1\st_1_ms1.feature</pre>
            <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2.feature</pre>
            <pre>C:\toppic_tutorial\tutorial_1\st_2_ms1.feature</pre>
            <pre>C:\toppic_tutorial\tutorial_1\st_2_ms2.feature</pre>
          </li>
          <li>Two XML files containing LC-MS features:
            <pre>C:\toppic_tutorial\tutorial_1\st_1_feature.xml</pre>
            <pre>C:\toppic_tutorial\tutorial_1\st_2_feature.xml</pre>
          </li>
          <li>Two folders containing deconvoluted MS/MS spectra in the JavaScript format
            <pre>C:\toppic_tutorial\tutorial_1\st_1_html\topfd</pre>
            <pre>C:\toppic_tutorial\tutorial_1\st_2_html\topfd</pre>
          </li>
          <p>
            The output files and folders can be downloaded
            <a href="./data_files/tutorial-1-msalign.zip">here</a>.
          </p>

          <h4 class="sectionTitle">4.5 Mass spectral identification using TopPIC</h4>
          <p>We use <b>toppic_gui</b> to search the MS/MS spectra in
            <code>st_1_ms2.msalign</code> and <code>st_2_ms2.msalign</code>
            against the protein database <code>uniprot-st.fasta</code> to
            identify PrSMs with a variable PTM file <code>var_mods.txt</code>,
            in which oxidation on methionine is set
            as a variable PTM. The variable PTM file can be downloaded
            <a href="./data_files/var_mods.txt">here</a>.
          </p>
          <ol class="textList">
            <li> Double click the executable file <code>toppic_gui.exe</code> in the folder
              <code>C:\toppic_tutorial\toppic</code>.
            </li>
            <li> Select <code>C:\toppic_tutorial\tutorial_1\uniprot-st.fasta</code> as the protein
              database file.</li>
            <li> Add <code>C:\toppic_tutorial\tutorial_1\st_1_ms2.msalign</code>
              and <code>C:\toppic_tutorial\tutorial_1\st_2_ms2.msalign</code> as
              mass spectrum data files. </li>
            <li> Input "combined" as the file name for combined identifications.</li>
            <li> Select <code>Carbamidomethylation on cysteine</code> as the fixed modification. </li>
            <li> Select file "var_mods.txt" as the variable PTM file. </li>
            <li> Check the checkbox <code>Decoy database</code>. </li>
            <li> Select <code>FDR</code> as the spectrum level cutoff type. </li>
            <li> Select <code>FDR</code> as the proteoform level cutoff type. </li>
            <li> Click to the button "Start"</li>
          </ol>
          <p>
            The screenshots of <code>toppic_gui</code> are shown below.
          </p>

          <p><img src="./images/toppic_2.png" height="400" width="600" style="border: #00008B 1px solid;" />
          </p>

          <p><img src="./images/toppic_3.png" height="400" width="600" style="border: #00008B 1px solid;" />
          </p>

          <p>For each input msalign file, TopPIC reports four TSV files, two XML files,
            and collections of HTML
            files for identified proteoforms. For example, the output files for st_1_ms2.msalign are
          <ul>
            <li>A TSV file containing identified PrSMs with a 1% spectrum-level FDR. When a proteoform is shared by
              multiple proteins, all the proteins are reported.
              <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_toppic_prsm.tsv</pre>
            </li>
            <li>A TSV file containing identified PrSMs with a 1% spectrum-level FDR. When a proteoform is shared by
              multiple proteins, only one protein is reported.
              <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_toppic_prsm_single.tsv</pre>
            </li>
            <li> A TSV file containing identified proteoforms and
              their best PrSMs with a 1% proteoform-level FDR. When a proteoform is shared by multiple proteins, all the
              proteins are reported.
              <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_toppic_proteoform.tsv</pre>
            </li>
            <li> A TSV file containing identified proteoforms and
              their best PrSMs with a 1% proteoform-level FDR. When a proteoform is shared by multiple proteins, only
              one protein is reported.
              <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_toppic_proteoform_single.tsv</pre>
            </li>
            <li> An XML file containing identified proteoforms and
              their best PrSMs with a 1% proteoform-level FDR:
              <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_toppic_proteoform.xml</pre>
            </li>
            <li> An XML file containing all identified PrSMs without clustering and filtering:
              <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_toppic_prsm.xml</pre>
            </li>
            <li>A folder containing JavaScript files of identified PrSMs with a 1%
              spectrum-level FDR:
              <pre>C:\toppic_tutorial\tutorial_1\st_1_html\toppic_prsm_cutoff</pre>
            </li>
            <li>
              A folder containing JavaScript files of identified PrSMs with a 1%
              proteoform-level FDR:
              <pre>C:\toppic_tutorial\tutorial_1\st_1_html\toppic_proteoform_cutoff</pre>
            </li>
            <li>
              A folder containing HTML files for the visualization of identified
              PrSMs:
              <pre>C:\toppic_tutorial\tutorial_1\st_1_html\topmsv</pre>
            </li>
          </ul>
          </p>
          <p>In addition, the identifications reported for st_1_ms2.msalign and
            st_2_ms2.msalign are combined, and filtered by a 1% spectrum-level FDR and
            a 1% proteoform-level FDR. The combined results are reported in the
            following files.
          <ul>
            <li>A TSV file containing combined PrSM identifications with a 1% spectrum level FDR. When a proteoform is
              shared by multiple proteins, all the proteins are reported.
              <pre>C:\toppic_tutorial\tutorial_1\combined_ms2_toppic_prsm.tsv</pre>
            </li>
            <li>A TSV file containing combined PrSM identifications with a 1% spectrum level FDR. When a proteoform is
              shared by multiple proteins, only one protein is reported.
              <pre>C:\toppic_tutorial\tutorial_1\combined_ms2_toppic_prsm_single.tsv</pre>
            </li>
            <li> A TSV file containing combined proteoform identifications and
              their best PrSMs with a 1% proteoform level FDR. When a proteoform is shared by multiple proteins, all the
              proteins are reported.
              <pre>C:\toppic_tutorial\tutorial_1\combined_ms2_toppic_proteoform.tsv</pre>
            </li>
            <li> A TSV file containing combined proteoform identifications and
              their best PrSMs with a 1% proteoform level FDR. When a proteoform is shared by multiple proteins, only
              one protein is reported.
              <pre>C:\toppic_tutorial\tutorial_1\combined_ms2_toppic_proteoform_single.tsv</pre>
            </li>
            <li> An XML file containing combined proteoform identifications and
              their best PrSMs with a 1% proteoform level FDR:
              <pre>C:\toppic_tutorial\tutorial_1\combined_ms2_toppic_proteoform.xml</pre>
            </li>
            <li> An XML file containing all identified PrSMs without clustering and filtering:
              <pre>C:\toppic_tutorial\tutorial_1\combined_ms2_toppic_prsm.xml</pre>
            </li>
            <li>A folder containing JavaScript files of combined PrSM
              identifications with a 1% spectrum-level FDR:
              <pre>C:\toppic_tutorial\tutorial_1\combined_html\toppic_prsm_cutoff</pre>
            </li>
            <li>
              A folder containing JavaScript files of combined PrSM identifications with a 1%
              proteoform-level FDR:
              <pre>C:\toppic_tutorial\tutorial_1\combined_html\toppic_proteoform_cutoff</pre>
            </li>
            <li>
              A folder containing HTML files for the visualization of identified
              PrSMs:
              <pre>C:\toppic_tutorial\tutorial_1\combined_html\topmsv</pre>
            </li>
          </ul>
          </p>

          <p> In the analysis, carbamidomethylation is selected as the fixed modification because
            proteins were reduced with dithiothreitol and alkylated with
            iodoacetamide before the MS experiment. When proteins are not reduced,
            no fixed modification should be selected.
          </p>

          <p>
            A shuffled decoy database is concatenated
            to the target database to estimate spectrum-level and proteoform-level
            FDRs. All identified PrSMs are first filtered by a
            1% spectrum-level FDR and the resulting PrSMs are reported in the
            file <code>combined_ms2_toppic_prsm.tsv</code>. The proteoforms corresponding to the PrSMs
            are further filtered using a 1% proteoform-level FDR and
            the resulting proteoforms and their corresponding best PrSMs are reported in the file
            <code>combined_ms2_toppic_proteoform.tsv</code>. Microsoft Excel can be used to open these two files.
            To browse the PrSM identifications,
            go to the folder <code>combined_html\topmsv</code> and use <b>Google
              Chrome</b> (Windows Edge and Firefox are not recommended)
            to open the file index.html.
          </p>

          <p>
            The output files can be downloaded
            <!-- <a href="filedownload.php?filename=./data_files/st-result.zip">here</a>.-->
            <a href="./data_files/tutorial-1-result.zip">here</a>.
            <!-- The running time of TopPIC is about 5 minutes. -->
          </p>

          <h4 class="sectionTitle">4.6 Data analysis using the command line interface</h4>

          <h4>4.6.1 Index file generation</h4>
          <p>We use <b>topindex</b> to generate index files from the protein database <code>uniprot-st.fasta</code>
            to speed up database search of TopPIC and TopMG.
          </p>

          <h4>File locations</h4>
          <ol class="textList">
            <li>Executable file:
              <pre>C:\toppic_tutorial\toppic\topindex.exe</pre>
            </li>
            <li>Input mzML file:
              <pre>C:\toppic_tutorial\tutorial_1\uniprot-st.fasta</pre>
            </li>
          </ol>
          <h4>Commands</h4>
          <div class="command-container" id="cmd1Container">
            <pre id="cmd1">cd C:\toppic_tutorial\tutorial_1</pre><button class="copyBtn"
              onclick="copyToClipboard('cmd1')">Copy</button>
          </div>
          <div class="command-container" id="cmd2Container">
            <pre id="cmd2">..\toppic\topindex -f C57 -d uniprot-st.fasta </pre><button class="copyBtn"
              onclick="copyToClipboard('cmd2')">Copy</button>
          </div>

          <h4>4.6.2 Mass spectral deconvolution</h4>

          <p>We use <b>topfd</b> for top-down mass spectral deconvolution.</p>

          <h4>File locations</h4>
          <ol class="textList">
            <li>Executable file:
              <pre>C:\toppic_tutorial\toppic\topfd.exe</pre>
            </li>
            <li>Input mzML file:
              <pre>C:\toppic_tutorial\tutorial_1\st_1.mzML</pre>
              <pre>C:\toppic_tutorial\tutorial_1\st_2.mzML</pre>
            </li>
          </ol>
          <h4>Commands</h4>
          <div class="command-container" id="cmd1Container">
            <pre id="cmd1">cd C:\toppic_tutorial\tutorial_1</pre><button class="copyBtn"
              onclick="copyToClipboard('cmd1')">Copy</button>
          </div>
          <div class="command-container" id="cmd2Container">
            <pre id="cmd2">..\toppic\topfd st_*.mzML</pre><button class="copyBtn"
              onclick="copyToClipboard('cmd2')">Copy</button>
          </div>
          <!--
    <p>The screenshot below shows the console output of the above command.</li>
    <p><img src="" height="250" width="500" border="1.5"/></p>
    -->
          <h4>4.6.3 Mass spectral identification using TopPIC</h4>

          <p>We use <b>toppic</b> to search the MS/MS spectra in <code>st_1_ms2.msalign</code>
            and <code>st_2_ms2.msalign</code>
            against the protein database <code>uniprot-st.fasta</code> to identify PrSMs.
          </p>
          <h4>File locations</h4>
          <ol class="textList">
            <li>Executable file:
              <pre>C:\toppic_tutorial\toppic\toppic.exe</pre>
            </li>
            <li>Input protein database file:
              <pre>C:\toppic_tutorial\tutorial_1\uniprot-st.fasta</pre>
            </li>
            <li>Input MS1 msalign files:
              <pre>C:\toppic_tutorial\tutorial_1\st_1_ms1.msalign</pre>
              <pre>C:\toppic_tutorial\tutorial_1\st_2_ms1.msalign</pre>
            </li>
            <li>Input MS/MS msalign files:
              <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2.msalign</pre>
              <pre>C:\toppic_tutorial\tutorial_1\st_2_ms2.msalign</pre>
            </li>
            <li>Input feature files:
              <pre>C:\toppic_tutorial\tutorial_1\st_1_ms1.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_1\st_2_ms1.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_1\st_2_ms2.feature</pre>
            </li>
            <li>Variable PTM file:
              <pre>C:\toppic_tutorial\tutorial_1\var_mods.txt</pre>
            </li>
          </ol>
          <h4>Commands</h4>
          <div class="command-container" id="cmd3Container">
            <pre id="cmd3">cd C:\toppic_tutorial\tutorial_1</pre>
            <button class="copyBtn" onclick="copyToClipboard('cmd3')">Copy</button>
          </div>
          <div class="command-container" id="cmd4Container">
            <pre
              id="cmd4">..\toppic\toppic -f C57 -d -t FDR -T FDR -b var_mods.txt -c combined uniprot-st.fasta st_*_ms2.msalign</pre>
            <button class="copyBtn" onclick="copyToClipboard('cmd4')">Copy</button>
          </div>
          <hr>

          <h3>5 Tutorial 2</h3>
          <p>
            We will use TopMG to analyze the data set st_1.raw described in Tutorial 1.
            <b>
              <font color=#FF0000> TopMG is still in the development stage. Please
                let us know if you find any bugs in it.</font>
            </b>.
          </p>
          <h4 class="sectionTitle">5.1 Data set and preprocessing </h4>
          The description of the data file and
          its preprocessing steps can be found in Sections 4.1 - 4.4.
          Click
          <a href="./data_files/tutorial-2-data.zip">here</a>
          to download the data files used in the
          analysis, save it in the folder <code>C:\toppic_tutorial\tutorial_2\</code>, and
          unzip it. It includes the following files.
          <ul>
            <li>A S. typhrimurium protein database file:
              <pre>C:\toppic_tutorial\tutorial_2\uniprot-st.fasta</pre>
            </li>
            <li> A deconvoluted MS1 data file:
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms1.msalign</pre>
            </li>
            <li> A deconvoluted MS/MS data file:
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms2.msalign</pre>
            </li>
            <li> Two MS feature file:
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms1.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms2.feature</pre>
            </li>
            <li>
              A text file containing a variable PTM: oxidation on methionine.
              <pre>C:\toppic_tutorial\tutorial_2\var_mods.txt</pre>
            </li>
            <li>
              A folder containing deconvoluted MS/MS spectra in the JavaScript format.
              <pre>C:\toppic_tutorial\tutorial_2\st_1_html\topfd</pre>
            </li>
          </ul>
          <h4 class="sectionTitle">5.2 Index file generation </h4>
          <p>
            To speed up database search, follow the steps in Section 4.2.1 to generate
            index files for the database file uniprot-st.fasta. If index files have
            been generated, it is not necessary to regenerate index files. You can
            copy the index folder to the folder C:\toppic_tutorial\tutorial_2\.
          </p>

          <h4 class="sectionTitle">5.3 Proteoform identification by TopMG </h4>
          <ol class="textList">
            <li> Double click the executable file <code>topmg_gui.exe</code> in the folder
              <code>C:\toppic_tutorial\toppic</code>.
            </li>
            <li> Select <code>C:\toppic_tutorial\tutorial_2\uniprot-st.fasta</code> as the protein
              database file.</li>
            <li> Add <code>C:\toppic_tutorial\tutorial_2\st_1_ms2.msalign</code> as a
              mass spectrum data file. </li>
            <li> Select <code>C:\toppic_tutorial\tutorial_2\variable_mods.txt</code> as the file of variable PTMs. </li>
            <li> Select <code>Carbamidomethylation on cysteine</code> as the fixed modification. </li>
            <li> Check the checkbox <code>Decoy database</code>. </li>
            <li> Select <code>FDR</code> as the spectrum level cutoff type. </li>
            <li> Set the spectrum level FDR cutoff to 0.05. </li>
            <li> Select <code>FDR</code> as the proteoform level cutoff type. </li>
            <li> Set the proteoform level FDR cutoff to 0.05. </li>
            <li> Click to the button "Start"</li>
          </ol>
          <p>
            The screenshots of <code>topmg_gui</code> are shown below.
          </p>

          <p><img src="./images/topmg_1.png" height="400" width="600" style="border: #00008B 1px solid;" />
          </p>

          <p><img src="./images/topmg_2.png" height="400" width="600" style="border: #00008B 1px solid;" />
          </p>

          <p>TopMG reports two TSV files, two XML files, and collections of HTML
            files for identified proteoforms.
          <ul>
            <li>A TSV file containing identified PrSMs with a 5% spectrum-level FDR. When a proteoform is shared by
              multiple proteins, all the proteins are reported.
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms2_topmg_prsm.tsv</pre>
            </li>
            <li>A TSV file containing identified PrSMs with a 5% spectrum-level FDR. When a proteoform is shared by
              multiple proteins, only one protein is reported.
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms2_topmg_prsm_single.tsv</pre>
            </li>
            <li> A TSV file containing identified proteoforms and
              their best PrSMs with a 5% proteoform-level FDR. When a proteoform is shared by multiple proteins, all the
              proteins are reported.
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms2_topmg_proteoform.tsv</pre>
            </li>
            <li> A TSV file containing identified proteoforms and
              their best PrSMs with a 5% proteoform-level FDR. When a proteoform is shared by multiple proteins, only
              one protein is reported.
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms2_topmg_proteoform_single.tsv</pre>
            </li>
            <li> An XML file containing identified proteoforms and
              their best PrSMs with a 5% proteoform-level FDR:
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms2_topmg_proteoform.xml</pre>
            </li>
            <li> An XML file containing all identified PrSMs without clustering and filtering:
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms2_topmg_prsm.xml</pre>
            </li>
            <li>A folder containing JavaScript files of identified PrSMs with a 5%
              spectrum-level FDR:
              <pre>C:\toppic_tutorial\tutorial_2\st_1_html\topmg_prsm_cutoff</pre>
            </li>
            <li>
              A folder containing JavaScript files of identified PrSMs with a 5%
              proteoform-level FDR:
              <pre>C:\toppic_tutorial\tutorial_2\st_1_html\topmg_proteoform_cutoff</pre>
            </li>
            <li>
              A folder containing HTML files for the visualization of identified
              PrSMs:
              <pre>C:\toppic_tutorial\tutorial_1\st_1_html\topmsv</pre>
            </li>
          </ul>
          </p>

          <p>
            The output files can be downloaded
            <!-- <a href="filedownload.php?filename=./data_files/st-result.zip">here</a>.-->
            <a href="./data_files/tutorial-2-result.zip">here</a>.
            <!-- The running time of TopPIC is about 5 minutes. -->
          </p>

          <p>
            To browse the PrSM identifications,
            go to the folder <code>st_1_html\topmsv</code> and use <b>Google
              Chrome</b> (Windows Edge and Firefox are not recommended)
            to open the file index.html.
          </p>


          <h4 class="sectionTitle">5.4 Data analysis using the command line
            interface</h4>


          <h4>File locations</h4>
          <ol class="textList">
            <li>Executable file:
              <pre>C:\toppic_tutorial\toppic\topmg.exe</pre>
            </li>
            <li>Input protein database file:
              <pre>C:\toppic_tutorial\tutorial_2\uniprot-st.fasta</pre>
            </li>
            <li>Input MS1 msalign file:
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms1.msalign</pre>
            </li>
            <li>Input MS/MS msalign file:
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms2.msalign</pre>
            </li>
            <li>MS feature files:
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms1.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_2\st_1_ms2.feature</pre>
            </li>
            <li>Variable PTM list:
              <pre>C:\toppic_tutorial\tutorial_2\var_mods.txt</pre>
            </li>
          </ol>
          <h4>Commands</h4>
          <div class="command-container" id="cmd5Container">
            <pre id="cmd5">cd C:\toppic_tutorial\tutorial_2 </pre>
            <button class="copyBtn" onclick="copyToClipboard('cmd5')">Copy</button>

          </div>

          <div class="command-container" id="cmd2Container">
            <pre id="cmd2">..\toppic\topindex -f C57 -d uniprot-st.fasta </pre><button class="copyBtn"
              onclick="copyToClipboard('cmd2')">Copy</button>
          </div>
          <div class="command-container" id="cmd6Container">
            <pre
              id="cmd6">..\toppic\topmg -f C57 -d -t FDR -v 0.05 -T FDR -V 0.05 -i variable_mods.txt uniprot-st.fasta st_1_ms2.msalign</pre>
            <button class="copyBtn" onclick="copyToClipboard('cmd6')">Copy</button>
          </div>

          <h3>6 Tutorial 3</h3>
          <p>
            We will use TopPIC and TopDiff to compare the abundance of
            proteoforms and find differentially expressed proteoforms
            using two MS data files of <i>Escherichia coli</i> cells (ecoli_1.raw
            and ecoli_2.raw).
          </p>
          <p>
            In the MS experiment, the protein extract of <i>E. coli</i>
            was reduced with dithiothreitol and alkylated with iodoacetamide.
            The protein mixture was separated by capillary zone
            electrophoresis and analyzed by
            an LTQ-Orbitrap mass spectrometer (Thermo Fisher Scientific).
            Technical duplicates were generated for testing proteoform
            quantification in two runs of the same sample.

            <!--MS and MS/MS spectra were collected at a resolution of 60,000 
      and 30,000, respectively. -->

          </p>
          <h4 class="sectionTitle">6.1 Data set and preprocessing </h4>
          The raw data files were processed following the steps found in Sections 4.1 - 4.4.
          Click
          <a href="./data_files/tutorial-3-data.zip">here</a>
          to download the data files used in the
          analysis, save it in the folder <code>C:\toppic_tutorial\tutorial_3\</code>, and
          unzip it. It includes the following files.
          <br>
          <br>
          <ul>
            <li>E.coli protein database file:
              <pre>C:\toppic_tutorial\tutorial_3\uniprot-ecoli.fasta</pre>
            </li>
            <li> Two mzML files:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_mzML</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_mzML</pre>
            </li>
            <li> Two deconvoluted MS1 data files:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms1.msalign</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms1.msalign</pre>
            </li>
            <li> Two deconvoluted MS/MS data files:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2.msalign</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms1.msalign</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2.msalign</pre>
            </li>
            <li> Four deconvoluted LC-MS feature files:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms1.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms1.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2.feature</pre>
            </li>
            <li> Two XML files containing LC-MS features:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_feature.xml</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_feature.xml</pre>
            </li>
            <li>
              Two folders containing deconvoluted MS/MS spectra in the JavaScript format.
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_html\topfd</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_html\topfd</pre>
            </li>
          </ul>

          <h4 class="sectionTitle">6.2 Index file generation </h4>
          <p>
            To speed up database search, follow the steps in Section 4.2.1 to generate
            index files for the database file uniprot-ecoli.fasta. If index files have
            been generated, it is not necessary to regenerate index files.
          </p>

          <h4 class="sectionTitle">6.3 Mass spectral identification using TopPIC </h4>
          <p>We use <b>toppic_gui</b> to search the MS/MS spectra in
            <code>ecoli_1_ms2.msalign</code> and <code>ecoli_2_ms2.msalign</code>
            against the protein database <code>uniprot-ecoli.fasta</code> to
            identify PrSMs.
          </p>
          <ol class="textList">
            <li> Double click the executable file <code>toppic_gui.exe</code> in the folder
              <code>C:\toppic_tutorial\toppic</code>.
            </li>
            <li> Select <code>C:\toppic_tutorial\tutorial_3\uniprot-ecoli.fasta</code> as the protein
              database file.</li>
            <li> Add <code>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2.msalign</code>
              and <code>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2.msalign</code> as
              mass spectrum data files. </li>
            <li> Select <code>Carbamidomethylation on cysteine</code> as the fixed modification. </li>
            <li> Check the checkbox <code>Decoy database</code>. </li>
            <li> Select <code>FDR</code> as the spectrum level cutoff type. </li>
            <li> Select <code>FDR</code> as the proteoform level cutoff type. </li>
            <li> Click to the button "Start"</li>
          </ol>
          <p>
            The screenshots of <code>toppic_gui</code> are shown below.
          </p>

          <p><img src="./images/toppic_4.png" height="400" width="600" style="border: #00008B 1px solid;" />
          </p>

          <p><img src="./images/toppic_5.png" height="400" width="600" style="border: #00008B 1px solid;" />
          </p>

          <p>For each input msalign file, TopPIC reports two TSV files, two XML files, and collections of html
            files for identified proteoforms. As a result, the output files for ecoli_1_ms2.msalign, ecoli_2_ms2.msalign
            are
          <ul>
            <li>Four TSV files containing identified PrSMs with a 1% spectrum-level FDR:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2_toppic_prsm.tsv</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2_toppic_prsm.tsv</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2_toppic_prsm_single.tsv</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2_toppic_prsm_single.tsv</pre>
            </li>
            <li> Four TSV files containing identified proteoforms and
              their best PrSMs with a 1% proteoform-level FDR:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2_toppic_proteoform.tsv</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2_toppic_proteoform.tsv</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2_toppic_proteoform_single.tsv</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2_toppic_proteoform_single.tsv</pre>
            </li>
            <li> Two XML files containing identified proteoforms and
              their best PrSMs with a 1% proteoform-level FDR:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2_toppic_proteoform.xml</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2_toppic_proteoform.xml</pre>
            </li>
            <li> Two XML files containing all identified PrSMs without clustering and filtering:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2_toppic_prsm.xml</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2_toppic_prsm.xml</pre>
            </li>
            <li>Two folders containing JavaScript files of identified PrSMs with a 1%
              spectrum-level FDR:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_html\toppic_prsm_cutoff</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_html\toppic_prsm_cutoff</pre>
            </li>
            <li>
              Two folders containing JavaScript files of identified PrSMs with a 1%
              proteoform-level FDR:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_html\toppic_proteoform_cutoff</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_html\toppic_proteoform_cutoff</pre>
            </li>
            <li>
              Two folders containing HTML files for the visualization of identified
              PrSMs:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_html\topmsv</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_html\topmsv</pre>
            </li>
          </ul>
          </p>

          <p>
            The output files can be downloaded
            <!-- <a href="filedownload.php?filename=./data_files/st-result.zip">here</a>.-->
            <a href="./data_files/tutorial-3-result.zip">here</a>.
            <!-- The running time of TopPIC is about 5 minutes. -->
          </p>


          <h4 class="sectionTitle">6.4 Proteoform abundance comparison by TopDiff </h4>
          <ol class="textList">
            <li> Double click the executable file <code>topdiff_gui.exe</code> in the folder
              <code>C:\toppic_tutorial\toppic</code>.
            </li>
            <li> Add <code>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2.msalign</code> and
              <code>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2.msalign</code> as
              mass spectrum data files.
            </li>
            <li> Click to the button "Start"</li>
          </ol>
          <p>
            The screenshots of <code>topdiff_gui</code> are shown below.
          </p>

          <p><img src="./images/topdiff_1.png" height="400" width="600" style="border: #00008B 1px solid;" />
          </p>

          <p>TopDiff reports one TSV file for identified proteoforms with their
            abundances in the input mass spectrum data
          <pre>C:\toppic_tutorial\tutorial_3\sample_diff.tsv</pre>
          </p>

          <p>
            The output file can be downloaded
            <!-- <a href="filedownload.php?filename=./data_files/st-result.zip">here</a>.-->
            <a href="./data_files/sample_diff.tsv">here</a>.
            <!-- The running time of TopPIC is about 5 minutes. -->
          </p>

          <h4 class="sectionTitle">6.5 Data analysis using the command line
            interface</h4>

          <h4 class="sectionTitle">6.5.1 Mass spectral identification by TopPIC</h4>

          <h4>File locations</h4>
          <ol class="textList">
            <li>Executable file:
              <pre>C:\toppic_tutorial\toppic\toppic.exe</pre>
            </li>
            <li>Input protein database file:
              <pre>C:\toppic_tutorial\tutorial_3\uniprot-ecoli.fasta</pre>
            </li>
            <li>MS1 msalign files:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms1.msalign</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms1.msalign</pre>
            </li>
            <li>MS/MS msalign files:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2.msalign</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2.msalign</pre>
            </li>
            <li>LC-MS feature files:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms1.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms1.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2.feature</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2.feature</pre>
            </li>
          </ol>
          <h4>Commands</h4>
          <div class="command-container" id="cmd7Container">
            <pre id="cmd7">cd C:\toppic_tutorial\tutorial_3 </pre>
            <button class="copyBtn" onclick="copyToClipboard('cmd7')">Copy</button>

          </div>

          <div class="command-container" id="cmd2Container">
            <pre id="cmd2">..\toppic\topindex -f C57 -d uniprot-ecoli.fasta </pre><button class="copyBtn"
              onclick="copyToClipboard('cmd2')">Copy</button>
          </div>

          <div class="command-container" id="cmd8Container">
            <pre id="cmd8">..\toppic\toppic -f C57 -d -t FDR -T FDR uniprot-ecoli.fasta ecoli_*_ms2.msalign</pre>
            <button class="copyBtn" onclick="copyToClipboard('cmd8')">Copy</button>
          </div>

          <h4 class="sectionTitle">6.5.2 Comparing proteoform abundances using
            TopDiff</h4>

          <h4>File locations</h4>
          <ol class="textList">
            <li>Executable file:
              <pre>C:\toppic_tutorial\toppic\topdiff.exe</pre>
            </li>
            <li>MS/MS msalign files:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2.msalign</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2.msalign</pre>
            </li>
            <li>Proteoform identification files:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_1_ms2_toppic_proteoform.xml</pre>
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_2_ms2_toppic_proteoform.xml</pre>
            </li>
          </ol>
          <h4>Commands</h4>
          <div class="command-container" id="cmd9Container">
            <pre id="cmd9">cd C:\toppic_tutorial\tutorial_3 </pre>
            <button class="copyBtn" onclick="copyToClipboard('cmd9')">Copy</button>

          </div>
          <div class="command-container" id="cmd10Container">
            <pre id="cmd10">..\toppic\topdiff ecoli_1_ms2.msalign ecoli_2_ms2.msalign</pre>
            <button class="copyBtn" onclick="copyToClipboard('cmd10')">Copy</button>
          </div>

          <h3>7 Tutorial 4</h3>
          In this tutorial, we will use TopDIA and TopPIC to analyze a
          top-down DIA MS data set of <i>E. coli</i> for proteoform
          identification.
          <h4 class="sectionTitle">7.1 Top-down MS/MS Dataset</h4>
          <p>
            In the MS experiment, the protein extract of <i>E. coli</i> was separated by
            an HPLC system and then analyzed with an Thermo Fusion Lumos mass
            spectrometer (Thermo Fisher Scientific) using the DIA mode. A total
            of six MS runs were carried out. In each run, quadrupole gas-phase
            fractionation was used to acquire MS1 scans for precursor ions in a
            fixed 80 m/z range. Specifically, the six runs covered the following
            m/z ranges: 720–800, 800–880, 880–960, 960–1040, 1040–1120, and
            1120–1200. MS1 spectra were collected with a resolution of 240,000
            (at 200 m/z), 4 micro scans, an automatic gain control (AGC) target
            value of 1 × 10<sup>6</sup>, and a maximum injection time of 200 ms. A 4 m/z
            isolation window was used to generate MS/MS spectra, resulting in a
            total of 20 MS/MS spectra for each cycle. MS/MS spectra were
            obtained with a scan range of 400–2000 m/z, a resolution of 60,000
            (at 200 m/z), 1 micro scan, an AGC target value of 1 ×
            10<sup>6</sup>, and a maximum injection time of 500 ms.
            Fragmentation was performed using higher-energy collisional
            dissociation (HCD) with 30% nominal collision energy (NCE). The MS
            run with an m/z range 800-880 was used in the tutorial
            (ecoli_800_880.raw).
          </p>
          <h4 class="sectionTitle">7.2 Data preprocessing</h4>
          <p>
            The method in Section 4.3 can be used to convert the raw file to its
            correponding centroided mzML file.
          </p>
          <p>Click <a href="./data_files/tutorial-4-data.zip">here</a>
            to download the data set, save it in the folder <code>C:\toppic_tutorial\tutorial_4\</code>, and unzip it in
            the same folder. It includes the following files.
          </p>
          <ul>
            <li>E.coli protein database file:
              <pre>C:\toppic_tutorial\tutorial_4\uniprot-ecoli.fasta</pre>
            </li>
            <li>One raw file:
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880.raw</pre>
            </li>
            <li> One mzML file:
              <pre>C:\toppic_tutorial\tutorial_3\ecoli_800_880.mzML</pre>
            </li>
          </ul>

          <h4 class="sectionTitle">7.3 Generating pseudo MS/MS spectra using TopDIA</h4>
          <p>We use <b>topdia_gui</b> for top-down mass spectral deconvolution.</p>
          <ol class="textList">
            <li> Double click the executable file <code>topdia_gui.exe</code> in the folder
              <code>C:\toppic_tutorial\toppic</code>.
            </li>
            <li> Add the file <code>C:\toppic_tutorial\tutorial_4\ecoli_800_880.mzML</code>
              as the input file.</li>
            <li> Click to the button "Start" to deconvolute the file.</li>
          </ol>
          <p>
            The screenshot of <code>topdia_gui</code> is shown below.
          </p>

          <p><img src="./images/topdia_1.png" height="400" width="600" style="border: #00008B 1px solid;" />
          </p>

          <p>TopDIA reports 2 output files for each isolation
            window. In the data file, a total of 20
            isolation windows are used: [800, 804], [804, 808], ..., [876, 880]. 
            For example, the output files for the isolation window [800, 804]
            are:
          <ol class="textList">
            <li>A tsv file containing MS/MS features:
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_800.000000_ms2.csv</pre>
            </li>
            <li>A csv file containing MS/MS features in the BatMass feature
              format:
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_800.000000_frac_ms2.mzrt.csv</pre>
            </li>
          </ol>
            
          </p>

          <p>In addition, TopDIA reports the following output files for the MS data file.</p>

          <ol class="textList">
            <li>A msalign file containing deconvoluted MS1 spectra:
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_ms1.msalign</pre>
            </li>
            <li>A msalign file containing pseudo MS/MS spectra:
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_ms2.msalign</pre>
            </li>
            <li>A tsv file containing LC-MS features:
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_ms1.csv</pre>
            </li>
            <li>A csv file containing LC-MS features in the BatMass feature
              format:
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_frac_ms1.mzrt.feature</pre>
            </li>
            <li>A file containing LC-MS features in the xml format:
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_feature.xml</pre>
            </li>
            <li>A text file containg LC-MS features: 
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_ms1.feature</pre>
            </li>
            <li>A text file containg MS/MS features: 
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_ms2.feature</pre>
            </li>
            <li>A msalign file containg spectral deconvolution results of MS/MS
              spectra: 
              <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_ms2_raw.msalign</pre>
            </li>
            <li>A folder containing deconvoluted spectra in the JavaScript format
              <pre>C:\toppic_tutorial\tutorial_1\ecoli_800_880_html</pre>
            </li>

          </ol>
          <p>
          <p>
              The output files and folders can be downloaded
              <a href="./data_files/tutorial-4-msalign.zip">here</a>.
          </p>

          <h4 class="sectionTitle">7.4 Proteoform identification using TopPIC</h4>
          <p>We use <b>toppic_gui</b> to search the MS/MS spectra in
            <code>ecoli_800_880_ms2.msalign</code>
            against the protein database <code>uniprot-ecoli.fasta</code> to
            identify PrSMs.
          </p>
          <ol class="textList">
            <li> Double click the executable file <code>toppic_gui.exe</code> in the folder
              <code>C:\toppic_tutorial\toppic</code>.
            </li>
            <li> Select <code>C:\toppic_tutorial\tutorial_4\uniprot-ecoli.fasta</code> as the protein
              database file.</li>
            <li> Add <code>C:\toppic_tutorial\tutorial_4\ecoli_800_880_pseudo_ms2.msalign</code>
              as the mass spectrum data file. </li>
            <li> Check the checkbox <code>Decoy database</code>. </li>
            <li> Check the checkbox <code>MS1 feature is missing</code>. </li>
            <li> Select <code>FDR</code> as the spectrum level cutoff type. </li>
            <li> Select <code>FDR</code> as the proteoform level cutoff type. </li>
            <li> Click to the button "Start"</li>
          </ol>
          <p>
            The screenshots of <code>toppic_gui</code> are shown below.
          </p>

          <p><img src="./images/tutorial_4_toppic_1.png" height="400" width="600" style="border: #00008B 1px solid;" />
          </p>

          <p><img src="./images/tutorial_4_toppic_2.png" height="400" width="600" style="border: #00008B 1px solid;" />
          </p>
          <p>The prsm identifications are stored in the file
          <code>ecoli_800_880_ms2_prsm_single.tsv</code>, and
            the proteoform identifications are stored in the file
            <code>ecoli_800_880_ms2_protoeform_single.tsv</code>.
            Detailed information of the output files of TopPIC can be found in
            Section 4.5.
          </p>
          <p>
              The output files and folders can be downloaded
              <a href="./data_files/tutorial-4-id.zip">here</a>.
          </p>

          <h4 class="sectionTitle">7.5 Data analysis using the command line interface</h4>

          <h3>7.5.2 Generating pseudo MS/MS spectra</h4>

            <p>We use <b>topdia</b> for the generation of pseudo MS/MS spectra.</p>

            <h4>File locations</h4>
            <ol class="textList">
              <li>Executable file:
                <pre>C:\toppic_tutorial\toppic\topdia.exe</pre>
              </li>
              <li>Input mzML file:
                <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880.mzML</pre>
              </li>
            </ol>
            <h4>Commands</h4>
            <div class="command-container" id="cmd1Container">
              <pre id="cmd1">cd C:\toppic_tutorial\tutorial_4</pre><button class="copyBtn"
                onclick="copyToClipboard('cmd1')">Copy</button>
            </div>
            <div class="command-container" id="cmd2Container">
              <pre id="cmd2">..\toppic\topdia ecoli_800_880.mzML</pre><button class="copyBtn"
                onclick="copyToClipboard('cmd2')">Copy</button>
            </div>
            <h3>7.5.3 Mass spectral identification using TopPIC</h4>

              <p>We use <b>toppic</b> to search the MS/MS spectra in <code>ecoli_800_880_ms2.msalign</code>
                against the protein database <code>uniprot-ecoli.fasta</code> to identify PrSMs.
              </p>
              <h4>File locations</h4>
              <ol class="textList">
                <li>Executable file:
                  <pre>C:\toppic_tutorial\toppic\toppic.exe</pre>
                </li>
                <li>Input protein database file:
                  <pre>C:\toppic_tutorial\tutorial_4\uniprot-ecoli.fasta</pre>
                </li>
                <li>Input MS/MS msalign files:
                  <pre>C:\toppic_tutorial\tutorial_4\ecoli_800_880_ms2.msalign</pre>
                </li>
              </ol>
              <h4>Commands</h4>
              <div class="command-container" id="cmd3Container">
                <pre id="cmd3">cd C:\toppic_tutorial\tutorial_4</pre>
                <button class="copyBtn" onclick="copyToClipboard('cmd3')">Copy</button>
              </div>
              <div class="command-container" id="cmd4Container">
                <pre
                  id="cmd4">..\toppic\toppic -x -d -t FDR -T FDR -x uniprot-ecoli.fasta ecoli_800_880_pseudo_ms2.msalign</pre>
                <button class="copyBtn" onclick="copyToClipboard('cmd4')">Copy</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>

  <script src="assets/js/jquery-1.7.1.min.js"></script>
  <script>
    //adding a button to copy command line commands
    function copyToClipboard(cmdId) {
      const copyText = document.getElementById(cmdId).textContent;
      const textArea = document.createElement('textarea');

      textArea.textContent = copyText;
      document.body.append(textArea);
      textArea.select();
      document.execCommand("copy");

      //show tooltip that shows the text is copied
      initializeTooltip(cmdId);

      //remove the text area
      removeTextArea(textArea);

    }
    function removeTextArea(textArea) {//removing the textarea that was created to just for copying purpose
      textArea.remove();
    }

    function initializeTooltip(cmdId) {
      const tooltip = document.createElement('div');

      document.getElementById(cmdId + "Container").appendChild(tooltip);
      tooltip.classList.add("copyBtnTooltip");
      tooltip.textContent = "Copied to clipboard";

      setTimeout(function resetToolTip() {//give time for a tooltip to appear and then completely disappear
        tooltip.style.opacity = "0";
        tooltip.style.transition = "opacity 1s";

        setTimeout(function removeToolTip() {
          tooltip.remove();
        }, 500);
      }, 200);
    }


  </script>


</body>

</html>
