<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>TopPIC Suite</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="description" content="">
<meta name="author" content="xwliu" >

<link href="assets/css/bootstrap.css" rel="stylesheet">
<link href="style.css" rel="stylesheet">
<link href="assets/css/bootstrap-responsive.css" rel="stylesheet">
<link rel="shortcut icon" href="toppic.ico">
</head>

<body style="overflow-y:scroll">

<div class="container">

  <div class="masthead">
    <h2 class="muted"><font color=#0066DD>TopPIC Suite</font></h2>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container">
          <ul class="nav">
            <li><a href="index.html">Summary</a></li>
            <li><a href="register.html">Downloads</a></li>
            <li><a href="manual.html">Manual</a></li>
            <li><a href="tutorial.html">Tutorial</a></li>
            <li><a href="publications.html">Publications</a></li>
            <li><a href="contact.html">Contact</a></li>
          </ul>
        </div>
      </div>
    </div><!-- /.navbar -->
  </div><!-- /.masthead -->

  <div class="row-fluid" style="font-size:16px">

    <div class="span-12">

      <h3>1 Overview</h3>
      In this tutorial, we use TopPIC suite to analyze several top-down MS/MS data sets
      on a computer with a Windows 10 operating system.
      <!-- <b>Annotated proteoform spectrum matches (PrSMs) identified from the data set can be browsed 
        <a href="data_files/st_ms2_prsm_cutoff_html/proteins.html">here</a></b>. -->

      <h3>2 Folders</h3>
      <p>Create the folders below for software packages and data sets
      used in this tutorial. 
      </p>
      <ol>
        <li>Create a new folder named <code>toppic_tutorial</code> on the <code>C:</code> drive of your system. 
        <li>Create a new subfolder named <code>toppic</code> in the folder <code>C:\toppic_tutorial\</code> for 
        the software TopPIC suite. 
        <li>Create a new subfolder named <code>tutorial_1</code> in the folder <code>C:\toppic_tutorial\</code>.
        <li>Create a new subfolder named <code>tutorial_2</code> in the folder <code>C:\toppic_tutorial\</code>.
        <li>Create a new subfolder named <code>tutorial_3</code> in the folder <code>C:\toppic_tutorial\</code>.    
        <!--
        <li>Create a new subfolder named 'results' in the folder C:\toppic_tutorial\ for saving
        data files. -->
      </ol>
      <p>The resulting folder structure is shown in the screenshot below.</p>
      <br>
      <p>
      <img src="./images/folder_1.png" height="200" width="600" style="border:
      #00008B 1px solid;"/>
      </p>

	   <h3>3 Software tools</h3>

      <h4>3.1 Msconvert</h4>
      <p>
      Msconvert is a software tool in <B>ProteoWizard</B> that converts raw
      files into various spectrum file formats. 

      <h4>3.1.1 Required software packages to install ProteoWizard</font></h4>
      <a href="https://www.microsoft.com/en-us/download/details.aspx?id=17851">
      Microsoft .NET Framework 4.0</a></li> or a higher version      
      is required. If you are using Windows 10, 
      Microsoft .NET Framework 4.0 has been preinstalled.

      <h4>3.1.2 Download ProteoWizard</h4>
      <p>
      Follow the steps to below download ProteoWizard:
      </p>
      <p>
      <ol>
        <li>Go to the link  <a href = "http://proteowizard.sourceforge.net/downloads.shtml">ProteoWizard</a>.</li>
        <li>Choose the type 'Windows 64-bit installer' for end users and download ProteoWizard.</li>
        <li>Double click the downloaded file pwiz-setup-3.0-x86-64.msi to install it. </li> 
      </ol>

      <h4>3.2 TopPIC suite</h4>

      <h4>3.2.1 Download TopPIC suite</h4>
      <p>
      <ol>
        <li>Go to the <a href = "./register.html">download webpage</a> of TopPIC suite.</li>
        <li>Choose the download type "Windows 64-bit zip file", fill out the
        registration form, and click "I accept license agreement and download
        TopPIC suite"
        to download it.</li>
        <li>Save it to the folder <code>C:\toppic_tutorial\toppic\</code>.</li>
        <li>Extract all the files of the downloaded zip file to the folder
        <code>C:\toppic_tutorial\toppic\</code>.</li> 
        <br>
        <p><img src="./images/toppic_1.png" height="200" width="600" 
        style="border: #00008B 1px solid;"/>
        </p>
      </ol>

      <h3>4 Tutorial 1</h3>
		In tutorial 1, we will use TopFD and TopPIC to analyze a top-down MS/MS data set of S. typhimurium 
		for proteoform identification. 
      <h4>4.1 Top-down MS/MS Date set</h4>
      <p>
      In the experiment, the protein extract of Salmonella typhimurium were reduced with dithiothreitol and alkylated with iodoacetamide. 
      The protein mixture was first separated by gas-phase fractionation, resulting in 7 fractions. 
      Each fraction was separated with an HPLC system coupled to an LTQ-Orbitrap mass spectrometer (Thermo Fisher Scientific). 
      MS and MS/MS spectra were collected at a resolution of 60,000 and 30,000, respectively. In this tutorial, we use the data files 
      of only two fractions (st_1.raw and st_2.raw) to speedup the analysis.  
      </p>
      <p>Click <a href="./data_files/tutorial-1-raw.zip">here</a>
      <!-- <p>Click <a
           href="filedownload.php?filename=./data_files/st.raw">here</a> -->
      to download the data set, save it in the folder <code>C:\toppic_tutorial\tutorial_1\</code>, and unzip it in the same folder.</p>

      <h4>4.2 Protein sequence database</h4>
      <p>A S. typhimurium <a href="https://www.uniprot.org/uniprot/?query=proteome:UP000001014%20reviewed:yes">proteome database</a> 
      was downloaded from the UniProt database. It contains 1,795 protein sequences.      
      </p>

      <p>Click <a href="filedownload.php?filename=./data_files/uniprot-st.fasta">here</a>
      to download the protein database and save it in the folder
      <code>C:\toppic_tutorial\tutorial_1\</code>.</p>

      <p>The folder <code>C:\toppic_tutorial\tutorial_1\</code> is shown in the screenshot below.</p>
      <br>
      <img src="./images/folder_2.png" height="200" width="600" 
      style="border: #00008B 1px solid;"/></p>
      </p>

      <h4>4.3 File format conversion</h4>

      <p>We use <b>MSConvertGUI</b> to convert the raw files st_1.raw and st_2.raw to mzML
      files.</p>
      <ol>
        <li> Search "msconvert" in the search box on the task bar of Windows 10, and run the desktop app "MSConvert."</li>
        <li> Add the files <code>c:\toppic_tutorial\tutorial_1\st_1.raw</code> and <code>c:\toppic_tutorial\tutorial_1\st_2.raw</code> as input files.</li>
        <li> Add the filter "peakPeaking vendor msLevel=1-" <b><font
            color=#FF0000>(important)</font></b>.</li>
        <li> Click to the button "Start" to perform file format conversion.</li>
      </ol>
      <p>
      The screenshot of MSConvertGUI is shown below. 
      </p>
      <p><img src="./images/msconvert_1.png" height="400" width="600" 
      style="border: #00008B 1px solid;"/>
      </p>

      <p>
      In the above file format conversion, the peak picking filter (step 3) is used to
      generate centroid, not profile, mzML data files, which are required by the
      spectral deconvolution tool TopFD. 
      </p>

      <p>
      The resulting output mzML files are
      <pre>C:\toppic_tutorial\tutorial_1\st_1.mzML</pre> and
      <pre>C:\toppic_tutorial\tutorial_1\st_2.mzML</pre>
      The sizes of the files are about 170
      MB and 194 MB, respectively. They can be downloaded <a
      href="filedownload.php?filename=./data_files/tutorial-1-mzml.zip">here</a>.  
      The
    running time for the file format conversion is less than one minute.   
      </p>

      <h4>4.4 Mass spectral deconvolution</h4>

      <p>We use <b>topfd_gui</b> for top-down mass spectral deconvolution.</p>
      <ol>
        <li> Double click the executable file <code>topfd_gui.exe</code> in the folder
        <code>C:\toppic_tutorial\toppic</code>.</li>
        <li> Add the file <code>c:\toppic_tutorial\tutorial_1\st_1.mzML</code> 
        and <code>c:\toppic_tutorial\tutorial_1\st_2.mzML</code> as input files.</li>
        <li> Click to the button "Start" to deconvolute the file.</li>
      </ol>
      <p>
      The screenshot of <code>topfd_gui</code> is shown below. 
      </p>

      <p><img src="./images/topfd_1.png" height="400" width="600" 
      style="border: #00008B 1px solid;"/>
      </p>

      TopFD reports six text files. 
      <ol>
        <li>Two msalign files containing deconvoluted MS1 spectra:
        <pre>C:\toppic_tutorial\tutorial_1\st_1_ms1.msalign</pre>
        <pre>C:\toppic_tutorial\tutorial_1\st_2_ms1.msalign</pre></li>
        <li>Two msalign files containing deconvoluted MS/MS spectra:
        <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2.msalign</pre>
        <pre>C:\toppic_tutorial\tutorial_1\st_2_ms2.msalign</pre></li>
        <li>Two text files containing LC/MS features of MS/MS spectra:
        <pre>C:\toppic_tutorial\tutorial_1\st_1.feature</pre>
        <pre>C:\toppic_tutorial\tutorial_1\st_2.feature</pre></li>
      </ol>
      <p>
      The six output files can be downloaded
      <a href="filedownload.php?filename=./data_files/tutorial-1-msalign.zip">here</a>.  
      <!--The running time of TopFD is about 6 minutes. -->
      </p>

      <h4>4.5 Mass spectral identification using TopPIC</h4>
      <p>We use <b>toppic_gui</b> to search MS/MS spectra in the files
      <code>st_1_ms2.msalign</code> and <code>st_2_ms2.msalign</code> 
      against the protein database <code>uniprot-st.fasta</code> to
      identify PrSMs. 
      </p>
      <ol>
        <li> Double click the executable file <code>toppic_gui.exe</code> in the folder
        <code>C:\toppic_tutorial\toppic</code>.</li>
        <li> Select <code>C:\toppic_tutorial\tutorial_1\uniprot-st.fasta</code> as the protein
        database file.</li>
        <li> Add <code>C:\toppic_tutorial\tutorial_1\st_1_ms2.msalign</code> 
        and <code>C:\toppic_tutorial\tutorial_1\st_2_ms2.msalign</code> as 
        mass spectrum data files. </li>
        <li> Input "combined" as the file name for combined identifications.</li>
        <li> Select <code>C57</code> as the fixed modification. </li>
        <li> Check the checkbox <code>Decoy database</code>. </li>
        <li> Select <code>FDR</code> as the spectrum level cutoff type. </li>
        <li> Select <code>FDR</code> as the proteoform level cutoff type. </li>
        <li> Click to the button "Start"</li>
      </ol>
      <p>
      The screenshot of <code>toppic_gui</code> is shown below. 
      </p>

      <p><img src="./images/toppic_2.png" height="400" width="600" style="border: #00008B 1px solid;"/>
      </p>

      <p><img src="./images/toppic_3.png" height="400" width="600" style="border: #00008B 1px solid;"/>
      </p>

      <p>For each input msalign file, TopPIC reports two csv files, an xml file, and a collection of html
      files for identified proteoforms. For example, the output files for st_1_ms2.msalign are 
      <ul>
        <li>A csv file containing identified PrSMs with a 1% spectrum level FDR:
        <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_prsm.csv</pre> 
        </li>
        <li> A csv file containing identified proteoforms and
        their best PrSMs with a 1% proteoform level FDR:
        <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_proteoform.csv</pre> 
        </li>
        <li> An xml file containing identified proteoforms and
        their best PrSMs with a 1% proteoform level FDR:
        <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_proteoform.xml</pre> 
        </li>  
        <li>A folder containing html files for the
        annotation of identified PrSMs with a 1% spectrum
        level FDR: 
        <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_prsm_cutoff_html</pre> 
        </li>
        <li>
        A folder containing html files for the
        annotation of identified PrSMs with a 1%
        proteoform level FDR: 
        <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2_proteoform_cutoff_html</pre> 
        </li>
      </ul>
      </p>
      <p>In addition, the identifications reported for st_1_ms2.msalign and st_2_ms2.msalign are 
      combined and filtered by a 1% spectrum-level FDR and a 1% proteoform level FDR. The combined results are reported in the following files.  
      <ul>
        <li>A csv file containing combined PrSM identifications with a 1% spectrum level FDR:
        <pre>C:\toppic_tutorial\tutorial_1\combined_prsm.csv</pre> 
        </li>
        <li> A csv file containing combined proteoform identifications and
        their best PrSMs with a 1% proteoform level FDR:
        <pre>C:\toppic_tutorial\tutorial_1\combined_proteoform.csv</pre> 
        </li>
        <li> An xml file containing combined proteoform identifications and
        their best PrSMs with a 1% proteoform level FDR:
        <pre>C:\toppic_tutorial\tutorial_1\combined_proteoform.xml</pre> 
        </li>  
        <li>A folder containing html files for the
        annotation of combined PrSM identifications with a 1% spectrum
        level FDR: 
        <pre>C:\toppic_tutorial\tutorial_1\combined_prsm_cutoff_html</pre> 
        </li>
        <li>
        A folder containing html files for the
        annotation of combined PrSM identifications with a 1%
        proteoform level FDR: 
        <pre>C:\toppic_tutorial\tutorial_1\combined_proteoform_cutoff_html</pre> 
        </li>
      </ul>
      </p>

      <p> In the analysis, C57 is selected as the fixed modification because
      proteins were reduced with dithiothreitol and alkylated with
      iodoacetamide before the MS experiment. When proteins are not reduced, C0
      should be selected.
      </p>

      <p>
      A shuffled decoy database is concatenated
      to the target database to estimate spectrum level and proteoform level
      FDRs. All identified PrSMs are first filtered by a
      1% spectrum level FDR and the resulting PrSMs are reported in the
      file <code>combined_prsm.csv</code>. The proteoforms corresponding to the PrSMs 
      are further filtered using a 1% proteoform level FDR and
      the resulting proteoforms and their corresponding best PrSMs are reported in the file
      <code>combined_proteoform.csv</code>. Microsoft Excel can be used to open these two files. 
      To browse the annotations in the two folders <code>combined_prsm_cutoff_html</code> and
      <code>combined_proteoform_cutoff_html</code>, use a web browser (Google Chrome is
      recommended) to open the file proteins.html.
      </p>

      <p>
      The output files can be downloaded
      <!-- <a href="filedownload.php?filename=./data_files/st-result.zip">here</a>.-->  
      <a href="./data_files/tutorial-1-result.zip">here</a>.  
      <!-- The running time of TopPIC is about 5 minutes. -->
      </p>
      
		<h4>4.6 Data analysis using the command line interface</h4>

      <h4>4.6.1 Mass spectral deconvolution</h4>

      <p>We use <b>topfd</b> for top-down mass spectral deconvolution.</p>

      <h4>File locations</h4>
      <ol>
        <li>Executable file: <pre>C:\toppic_tutorial\toppic\topfd.exe</pre></li>
        <li>Input mzML file: <pre>C:\toppic_tutorial\tutorial_1\st_1.mzML</pre>
        <pre>C:\toppic_tutorial\tutorial_1\st_2.mzML</pre>
        </li>
      </ol>
      <h4>Commands</h4>
      <pre>cd c:\toppic_tutorial\tutorial_1</pre>
      <pre>..\toppic\topfd st_*.mzML</pre>
      <!--
      <p>The screenshot below shows the console output of the above command.</li>
      <p><img src="" height="250" width="500" border="1.5"/></p>
      -->

		<h4>4.6.2 Mass spectral identification using TopPIC</font></h4>

      <p>We use <b>toppic</b> to search MS/MS spectra in the files <code>st_1_ms2.msalign</code>
      and <code>st_2_ms2.msalign</code>
      against the protein database <code>uniprot-st.fasta</code> to identify PrSMs. 
      </p>
      <h4>File locations</h4>
      <ol>
        <li>Executable file: <pre>C:\toppic_tutorial\toppic\toppic.exe</pre></li>
        <li>Input protein database file:
        <pre>C:\toppic_tutorial\tutorial_1\unprot-st.fasta</pre></li>
        <li>Input MS/MS msalign files: <pre>C:\toppic_tutorial\tutorial_1\st_1_ms2.msalign</pre>
        <pre>C:\toppic_tutorial\tutorial_1\st_2_ms2.msalign</pre>
        </li>
        <li>Input feature files: <pre>C:\toppic_tutorial\tutorial_1\st_1.feature</pre>
        <pre>C:\toppic_tutorial\tutorial_1\st_2.feature</pre>
        </li>
      </ol>
      <h4>Commands</h4>
      <pre>cd c:\toppic_tutorial\tutorial_1</pre>
      <pre>..\toppic\toppic -f C57 -d -t FDR -T FDR -c combined uniprot-st.fasta st_*_ms2.msalign</pre>
      <!--
      <p>The screenshot below shows the console output of the above command.</li>
      <p><img src="" height="250" width="500" border="1.5"/></p>
      -->

      <h3>5. Tutorial 2 </h3>
      <p>In this tutorial, we will use TopFD and TopPIC to identify proteoforms from top-down MS data generated from two 
      Salmonella typhimurium samples. Proteoform identified from the two samples 
      will be combined, and their abundances will be reported. 
		</p>      
      
      <h4>5.1 Top-down MS/MS data</h4>
		<p>Two Salmonella typhimurium samples were analyzed by top-down MS. One sample was grown in Luria-Bertani (LB) 
		medium (basal condition), and the other in 
		low phosphate low magnesium (LPM) acidic minimal medium (infection-like condition).       
      The protein extract of the each sample was separated by an RPLC system coupled with an LTQ Orbitrap Velos 
      mass spectrometer. MS1 and ETD MS/MS spectra 
      were collected at a resolution of 60,000.
      </p>
		<p>Click <a href="./data_files/tutorial-2-raw.zip">here</a>
      to download the data set, save it in the folder <code>C:\toppic_tutorial\tutorial_2\</code>, and unzip it in the same folder.</p>

      <h4>5.2 Protein sequence database</h4>
      <p>The S. typhimurium <a href="https://www.uniprot.org/uniprot/?query=proteome:UP000001014%20reviewed:yes">proteome database</a> 
      was downloaded from the UniProt database. It contains 1,795 protein sequences.      
      </p>
      <p>Click <a href="filedownload.php?filename=./data_files/uniprot-st.fasta">here</a>
      to download the protein database and save it in the folder
      <code>C:\toppic_tutorial\tutorial_2\</code>.</p>
            
      <h4>5.3 Data preprocessing</h4>
		<p>      
      Follow the steps in Sections 4.3 and 4.4 for data format conversion and spectral deconvolution. It may take 
      a couple of hours to deconvolute the spectra.  
		You can download the resulting files        
		<a href="./data_files/tutorial-2-preprocess.zip">here</a> directly.      
		</p>       
		
		<h4>5.4 Proteoform identification by TopPIC </h4>
		<p>We use <b>toppic_gui</b> to search MS/MS spectra in the files
      <code>st_lb_ms2.msalign</code> and <code>st_lpm_ms2.msalign</code> 
      against the protein database <code>uniprot-st.fasta</code> to
      identify PrSMs. 
      </p>
      <ol>
        <li> Double click the executable file <code>toppic_gui.exe</code> in the folder
        <code>C:\toppic_tutorial\toppic</code>.</li>
        <li> Select <code>C:\toppic_tutorial\tutorial_2\uniprot-st.fasta</code> as the protein
        database file.</li>
        <li> Add <code>C:\toppic_tutorial\tutorial_2\st_lb_ms2.msalign</code> 
        and <code>C:\toppic_tutorial\tutorial_2\st_lpm_ms2.msalign</code> as 
        mass spectrum data files. </li>
        <li> Leave the file name for combined identifications blank.</li>
        <li> Check the checkbox <code>Decoy database</code>. </li>
        <li> Select <code>FDR</code> as the spectrum level cutoff type. </li>
        <li> Select <code>FDR</code> as the proteoform level cutoff type. </li>
        <li> Click to the button "Start"</li>
      </ol>
      <p>
      The screenshot of <code>toppic_gui</code> is shown below. 
      </p>

      <p><img src="./images/toppic_4.png" height="400" width="600" style="border: #00008B 1px solid;"/>
      </p>

      <p><img src="./images/toppic_5.png" height="400" width="600" style="border: #00008B 1px solid;"/>
      </p>
      <p>The identification files can be downloaded
      <a href="./data_files/tutorial-2-result.zip">here</a>. 
      </p>
		<h4>5.5 Merging proteoform identifications </h4>
		<p>
      We use <b>topmerge_gui</b> to merge identification files <code>C:\toppic_tutorial_2\st_lb_ms2_toppic_proteoform.xml</code>
      and <code>C:\toppic_tutorial\st_lpm_ms2_toppic_proteoform.xml</code>.
      </p>
		<ol>
        <li> Double click the executable file <code>topmerge_gui.exe</code> in the folder
        <code>C:\toppic_tutorial\toppic</code>.</li>
        <li> Select <code>C:\toppic_tutorial\tutorial_2\uniprot-st.fasta</code> as the protein
        database file.</li>
        <li> Add <code>C:\toppic_tutorial\tutorial_2\st_lb_ms2_toppic_proteoform.xml</code> 
        and <code>C:\toppic_tutorial\tutorial_2\st_lpm_ms2_toppic_proteoform.xml</code> as 
        identification files. </li>
        <li> Click to the button "Start"</li>
      </ol>      
      <p>
      The screenshot of <code>topmerge_gui</code> is shown below. 
      </p>
      <p><img src="./images/topmerge_1.png" height="400" width="600" style="border: #00008B 1px solid;"/>
      </p>
      <p>TopMerge outputs a csv file <code>C:\toppic_tutorial\tutorial_2\sample_merged.csv</code>, which can be downloaded 
      <a href="./data_files/tutorial-2-merge.zip">here</a>.
      </p>
      
 
		<h4>5.6 Data analysis using the command line interface</h4>
		<p> See Section 4.6.1 for the commands for spectral deconvolution. </p>

		<h4>5.6.1 Mass spectral identification using TopPIC</font></h4>

      <p>We use <b>toppic</b> to search MS/MS spectra files <code>st_lb_ms2.msalign</code>
      and <code>st_lpm_ms2.msalign</code>
      against the protein database <code>uniprot-st.fasta</code> to identify PrSMs. 
      </p>
      <h4>File locations</h4>
      <ol>
        <li>Executable file: <pre>C:\toppic_tutorial\toppic\toppic.exe</pre></li>
        <li>Input protein database file:
        <pre>C:\toppic_tutorial\tutorial_2\unprot-st.fasta</pre></li>
        <li>Input MS/MS msalign files: <pre>C:\toppic_tutorial\tutorial_2\st_lb_ms2.msalign</pre>
        <pre>C:\toppic_tutorial\tutorial_2\st_lpm_ms2.msalign</pre>
        </li>
        <li>Input feature files: <pre>C:\toppic_tutorial\tutorial_2\st_lb.feature</pre>
        <pre>C:\toppic_tutorial\tutorial_2\st_lpm.feature</pre>
        </li>
      </ol>
      <h4>Commands</h4>
      <pre>cd c:\toppic_tutorial\tutorial_2</pre>
      <pre>..\toppic\toppic d -t FDR -T FDR uniprot-st.fasta st_*_ms2.msalign</pre>      
      
      <h4>5.6.2 Merging proteoform identifications using TopMerge</font></h4>

      <h4>File locations</h4>
      <ol>
        <li>Executable file: <pre>C:\toppic_tutorial\toppic\topmerge.exe</pre></li>
        <li>Input protein database file:
        <pre>C:\toppic_tutorial\tutorial_2\unprot-st.fasta</pre></li>
        <li>Proteoform identification files: <pre>C:\toppic_tutorial\tutorial_2\st_lb_ms2_toppic_proteoform.xml</pre>
        <pre>C:\toppic_tutorial\tutorial_2\st_lpm_ms2_toppic_proteoform.xml</pre>
      </ol>
      <h4>Commands</h4>
      <pre>cd c:\toppic_tutorial\tutorial_2</pre>
      <pre>..\toppic\topmerge uniprot-st.fasta st_*_ms2_toppic_proteoform.xml</pre>   
      
      
      
      <h3>6 Tutorial 3</h3>
      <p>
      We will use TopMG to analyze the data set st_1.raw described in Tutorial 1. 
		</p>
		<h4>6.1 Data set and preprocessing </h4>      
      The description of the data file and 
      its preprocessing steps can be found in Sections 4.1 - 4.4. 
      Click <a href="./data_files/tutorial-3-data.zip">here</a> to download data files used in the 
      analysis, save it in the folder <code>C:\toppic_tutorial\tutorial_3\</code>, and 
      unzip it. It includes the following files. 
      <ul>
        <li>A S. typhrimurium protein database file:
        <pre>C:\toppic_tutorial\tutorial_3\uniprot-st.fasta</pre> 
        </li>
        <li> A deconvoluted MS/MS data file:
        <pre>C:\toppic_tutorial\tutorial_3\st_1_ms2.msalign</pre> 
        </li> A MS1 feature file:  
        <pre>C:\toppic_tutorial\tutorial_3\st_1.feature</pre> 
        </li>
        <li>
        A text file containing two variable PTMs: oxidation and methylation.  
        <pre>C:\toppic_tutorial\tutorial_3\variable_ptms.txt</pre> 
        </li>
      </ul>
      
      <h4>6.2 Proteoform identification by TopMG </h4>
      <ol>
        <li> Double click the executable file <code>topmg_gui.exe</code> in the folder
        <code>C:\toppic_tutorial\toppic</code>.</li>
        <li> Select <code>C:\toppic_tutorial\tutorial_3\uniprot-st.fasta</code> as the protein
        database file.</li>
        <li> Add <code>C:\toppic_tutorial\tutorial_3\st_1_ms2.msalign</code> as a
        mass spectrum data file. </li>
        <li> Select <code>C57</code> as the fixed modification. </li>
        <li> Check the checkbox <code>Decoy database</code>. </li>
        <li> Select <code>FDR</code> as the spectrum level cutoff type. </li>
        <li> Set the spectrum level FDR cutoff to 0.05. </li>
        <li> Select <code>FDR</code> as the proteoform level cutoff type. </li>
        <li> Set the proteoform level FDR cutoff to 0.05. </li>
        <li> Select <code>C:\toppic_tutorial\tutorial_3\variable_mods.txt</code> as the modification file. </li>
        <li> Click to the button "Start"</li>
      </ol>
      <p>
      The screenshots of <code>topmg_gui</code> are shown below. 
      </p>

      <p><img src="./images/topmg_1.png" height="400" width="600" style="border: #00008B 1px solid;"/>
      </p>

      <p><img src="./images/topmg_2.png" height="400" width="600" style="border: #00008B 1px solid;"/>
      </p>

      <p>TopMG reports two cvs files, an xml file, and a collection of html
      files for identified proteoforms. 
      <ul>
        <li>A csv file containing identified PrSMs with a 5% spectrum level FDR:
        <pre>C:\toppic_tutorial\tutorial_3\st_1_ms2_topmg_prsm.csv</pre> 
        </li>
        <li> A csv file containing identified proteoforms and
        their best PrSMs with a 5% proteoform level FDR:
        <pre>C:\toppic_tutorial\data\st_1_ms2_topmg_proteoform.csv</pre> 
        </li>
        <li> An xml file containing identified proteoforms and
        their best PrSMs with a 5% proteoform level FDR:
        <pre>C:\toppic_tutorial\data\st_1_ms2_topmg_proteoform.xml</pre> 
        </li> 
        <li>A folder containing html files for the
        annotation of identified PrSMs with a 5% spectrum
        level FDR: 
        <pre>C:\toppic_tutorial\data\st_1_ms2_prsm_cutoff_html</pre> 
        </li>
        <li>
        A folder containing html files for the
        annotation of identified PrSMs with a 5%
        proteoform level FDR: 
        <pre>C:\toppic_tutorial\data\st_1_ms2_proteoform_cutoff_html</pre> 
        </li>
      </ul>
      </p>

      <p>
      The output files can be downloaded
      <!-- <a href="filedownload.php?filename=./data_files/st-result.zip">here</a>.-->  
      <a href="./data_files/tutorial-3-result.zip">here</a>.  
      <!-- The running time of TopPIC is about 5 minutes. -->
      </p>

      <h4>6.3 Data analysis using the command line interface</h3>


      <h4>File locations</h4>
      <ol>
        <li>Executable file: <pre>C:\toppic_tutorial\toppic\topmg.exe</pre></li>
        <li>Input protein database file:
        <pre>C:\toppic_tutorial\tutorial_3\unprot-st.fasta</pre></li>
        <li>Input MS/MS msalign file: <pre>C:\toppic_tutorial\tutorial_3\st_1_ms2.msalign</pre></li>
        <li>MS feature file: <pre>C:\toppic_tutorial\tutorial_3\st_1.feature</pre></li>
        <li>Variable PTM list: <pre>C:\toppic_tutorial\tutorial_3\variable_mods.txt</pre></li>
      </ol>
      <h4>Commands</h4>
      <pre>cd c:\toppic_tutorial\tutorial_3</pre>
      <pre>..\toppic\topmg -f C57 -d -t FDR -v 0.05 -T FDR -V 0.05 -i variable_mods.txt uniprot-st.fasta st_1_ms2.msalign</pre>
    </div>

  </div>
</div>
<script src="assets/js/jquery-1.7.1.min.js"></script>

</body>
</html>
